<body bgcolor=#FFFFFF>

<a NAME="top"></a>

<h2>MTB Background</h2>
<ul>
  <li><a href="#data">What information can I find in the Mouse Tumor Biology Database?</a></li>
  <li><a href="#submitting">How can I contribute my data to MTB?</a></li>
  <li><a href="#citing">How do I cite MTB?</a></li>
  <li><a href="#error">What should I do if I get an error message?</a></li>
</ul>

<h2>How to query the Mouse Tumor Biology Database...</h2>
<ul>
  <li><a href="#queries">Getting started, General Search Tips</a></li>
  
  <ul>
    <li><a href="#sort">How do the sort options work?</a></li>
    <li><a href="#number">How do I change the number of records returned by a search?</a></li>
    <li><a href="#text">How do I effectively use text string search boxes?</a></li>
    <li><a href="#pick">How do I effectively use pick list searching?</a></li>
    <li><a href="#path">How do I access Pathology Images in the database?</a></li>
    <li><a href="#QTL">How do I use the Cancer QTL Viewer</a></li>
    <li><a href="#dynamicgrid">How do I use the Tumor Frequency Grid</a></li>
  </ul>
</ul>

<h2>What values are acceptable in each of the query form fields?</h2>
<ul>
  <li>Query form and field descriptions for the Mouse Tumor Biology Database. Example queries follow each subsection.</li>
  <ul>
    <li><a href="#searchbox">"Search for" Search Box</a></li>
    <li><a href="#quickorgan">Quick Organ/Tissue Search</a></li>
    <li><a href="#tumor">Tumor</a></li>
    <li><a href="#strain">Strain</a></li>
    <li><a href="#genetics">Genetics</a></li>
    <li><a href="#humangene">Human Gene</a></li>
    <li><a href="#pathology">Pathology Images</a></li>
    <li><a href="#reference">Reference</a></li>
    <li><a href="#advanced">Advanced</a></li>
  </ul>
</ul>

<h2>How do I interpret my search results?</h2>
<ul>
  <li><a href="#interpreting">General tips on interpreting search results.</a></li>
  <li><a href="#tumorresults">Tumor Search Results</a></li>
  <li><a href="#tumorsummary">Tumor Summary</a></li>
  <li><a href="#strainresults">Strain Search Results</a></li>
  <li><a href="#straindetail">Strain Tumor Overview</a></li>
  <li><a href="#geneticsresults">Genetics Search Results</a></li>
  <li><a href="#geneticchange">Genetic Change Summary</a></li>
  <li><a href="#cytogeneticsummary">Cytogenetic Summary</a></li>
  <li><a href="#humangeneresults">Human Gene Results</a></li>
  <li><a href="#pathresults">Pathology Image Search Results</a></li>
  <li><a href="#pathdetail">Pathology Image Detail</a></li>
  <li><a href="#referenceresults">Reference Search Results</a></li>
  <li><a href="#referencedetail">Reference Detail</a></li>
  <li><a href="#pdxsearch">PDX Search</a></li>
  <li><a href="#pdxdetails">PDX Details</a></li>
  <li><a href="#pdxcomparison">PDX Comparison Results</a></li>
</ul>

<h2>See Also:</h2>
<ul>
  <li><a href="http://www.informatics.jax.org/glossary" target="_blank">MGI Glossary</a></li>
</ul>

<center><a href="/">MTB Home Page</a></center>

<hr>

<a NAME="data"></a>
<h2>MTB Background</h2>
<h3>What information can I find in the Mouse Tumor Biology (MTB) Database?</h3>

The Mouse Tumor Biology Database (MTB) was designed to aid researchers in such areas as choosing experimental models, reviewing patterns of mutations in specific cancers,
and identifying genes that are commonly mutated across a spectrum of cancers.

<p>MTB supports the use of the mouse as a model system of hereditary cancer by providing electronic access to:

<ol><li> Descriptions of tumors (data are organized by strain and organ of tumor origin)</li>
  <li>Tumor frequency and latency data</li>
  
  <li>Information on genetic factors associated with tumors and tumor development</li>
  <li>Tumor pathology reports and images</li>
  <li>References, both published and unpublished</li>
  <li>Links to related online resources, including:</li>
  <ul><li><a href="http://www.informatics.jax.org/" target="_blank">Mouse Genome Informatics (MGI)</a></li>
    <li><a href="http://www.jax.org/phenome" target="_blank">Mouse Phenome Database (MPD)</a></li>
    <li><a href="http://jaxmice.jax.org/" target="_blank">JAX&reg; Mice</a></li>
    <li><a href="http://mouse.ncifcrf.gov/" target="_blank">NCI Mouse Repository</a></li>
  </ul>
</ol>

<h4>Important information for users of MTB</h4>
<ol>
  <li><b>MTB reports negative as well as positive data.</b> For example, strains of mice that are reported to have a zero frequency of a particular tumor type are included
  in the database.</li>
  <li><b>Tumor data in MTB are from studies of tumors that arise in mice either spontaneously or after administration of some sort of agent.</b> MTB does not include data on
  tumors that arise in humans or other species and are subsequently transplanted into mice.</li>
  <li><b>MTB is NOT current with the published literature.</b> Populating MTB is a high priority, but the volume of data available for mouse cancer biology is very large and
  it will take some time to enter the data. Descriptions of the data sources and data entry priorities for MTB are described below.</li>
  <li><b>The mouse strain information in MTB is NOT limited to the strains distributed by the Jackson Laboratory.</b> However, if a strain in MTB is distributed by The Jackson
  Laboratory, a link to the data sheet in the JAX&reg; Mice database is provided.</li>
  <li><b>The data in the Mouse Tumor Biology Database come from the following sources:</b></li>
  <ul><li>the published literature (current and retrospective),</li>
    <li>the submission of tumor pathology slides and/or images from investigators (from published or unpublished data), and</li>
    <li>tumor frequency and pathology data from routine animal health screenings of mouse colonies (often unpublished data).</li>
  </ul>
  <li><b>Priority for entering data in the database is determined by the novelty of the mouse model, the quality of the data, and the organ system involved.</b> The organ
    systems of high priority are those with the highest reported mortality rate among the United States population, according to the National Cancer Institute.
  </li>
</ol>

<p><a href="#top">Back to Top</a>
<center><hr width=80%></center>

<p>
<a NAME="submitting"></a>
<h3>How can I contribute my data to MTB?</h3>

Data in the Mouse Tumor Biology Database come both from the published scientific literature and from unpublished, routine animal health screenings of mouse colonies. Data
submissions from the scientific community are welcome by contacting
<a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a> for information on the data submission process.

<p><a href="#top">Back to Top</a>
<center><hr width=80%></center>

<a NAME="citing"></a>
<h3>How do I cite MTB?</h3>
MTB is supported by grant CA89713, entitled "Electronic Access to Mouse Tumor Data", awarded to Janan T. Eppig from the National Cancer Institute (NCI) of the National Institutes
of Health (NIH).
<p>Please use the following citation when referring to the Mouse Tumor Biology Database.
<ul><li><b>Krupke DM, Begley DA, Sundberg JP, Bult CJ, Eppig JT. 2008. The Mouse Tumor Biology database. Nat Rev Cancer 8(6):459-65.</b></li></ul>
<p>If you wish to cite a specific area of MTB we suggest a format similar to the following example:
<ul><li><b>Some tumor data for this paper were retrieved from the Mouse Tumor Biology Database (MTB), Mouse Genome Informatics, The Jackson Laboratory, Bar Harbor, Maine. 
World Wide Web (URL:http://www.informatics.jax.org/). (October, 1998 i.e., the date you retrieved the data cited).</b></li></ul>

<p><a href="#top">Back to Top</a>
<center><hr width=80%></center>

<p>
<a NAME="error"></a>
<h3>What should I do if I get an error message?</h3>
Here are some possible solutions to common problems. If you continue to experience problems, please contact <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User
support </a>.
<p>
<TABLE BORDER="1" CELLSPACING="2" CELLPADDING="2" WIDTH="100%">
  <TR><TH ALIGN="center" VALIGN="top">Problem</TH>
    <TH ALIGN="center" VALIGN="top" NOWRAP>Probable Cause</TH>
    <TH ALIGN="center" VALIGN="top">Solution</TH>
  </TR>
  <TR><TD ALIGN="lEFT" VALIGN="top">Zero records were returned.</TD>
    <TD ALIGN="lEFT" VALIGN="top">The combination of query terms does not match any of the records in the database.</TD>
    <TD ALIGN="align="lEFT VALIGN="top">Try using fewer search conditions to broaden your query.</TD>
  </TR>
  <TR><TD ALIGN="left" VALIGN="top">The results page is only displaying 25 results even though the search found many more than 25 records.</TD>
    <TD ALIGN="left" VALIGN="top">The default limit for the number of records returned was not changed on the query form.</TD>
    <TD ALIGN="left" VALIGN="top">Redo the query with the "Max number of items returned" set to a higher number.</TD>
  </TR>
  <TR><TD ALIGN="left" VALIGN="top">"Invalid URL<br>If you arrived here from a bookmark generated from a previous version of MTB, those URLs are no longer valid. Please perform
    your search again using the search forms to the left."</TD>
    <TD ALIGN="left" VALIGN="top">A bookmark or hyperlink saved from the previous version of MTB may have been used.</TD>
    <TD ALIGN="left" VALIGN="top">Update the bookmark or hyperlink to the current format.<br>If you need assistance searching the new version of MTB or coding URL links to
    specific data in the database please contact <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml" target="_blank">User Support</a>.</TD>
  </TR>
</TABLE>

<p><a href="#top">Back to Top</a>
<center><hr width=100%></center>

<a NAME="queries"></a></h4>

<h2>How to query the Mouse Tumor Biology Database</h2>

You begin the MTB query process by filling out one or more fields in a search form and clicking on the <b>"Search"</b> or <b>"Go"</b> buttons (depending on search form you use).
Your query results will be returned as a Search Results page with hyperlinks to additional information. A common query strategy is to start with a very general query, review
the results, and then go back to the search form and refine the query so that fewer records are returned.
To clear all fields of a search form, click <b>"Reset"</b>. This will reset all fields to their default values on the form.
<p>
<center><hr width=80%></center>

<h3>Getting Started, General Search Tips</h3>
<ul>
  <li>You can search the database using the Search Box, the Quick Organ Search, one of the search forms labeled 
  Tumor, Strain, Genetics, Pathology Images, References and Advanced, the Cancer QTL Viewer, or the Tumor Frequency Grid.</li>
  <li>Searching the database is made simple by using a combination of text string and pick list searches for specific terms.</li>
  <li>The following provides general information for MTB queries. Specific information and examples for individual search forms is given below.</li>
</ul>
<p>
<center><hr width=80%></center>

<a name="sort">
<h3></a>How do the sort options work?</h3>
<ul>
  <li>The sort option allows you to change the order in which records appear on your results page. The sort options are specific for each search form.</li>
</ul>
<p>
<center><hr width=80%></center>

<a NAME="number"></a>
<h3><a name="sort"></a>How can I change the number of records returned by a search?</h3>
<ul>
  <li>The default number of records returned by a search is 25 for Tumor, Strain, and Genetics searches, and 5 for Pathology Image searches. By changing the <b>"Max number of
    items returned"</b> in the search form, you change the number of records shown on the subsequent Search results page. To view <b>ALL</b> records matching your search criteria,
  choose the <b>"No Limit"</b> option.</li>
</ul>
<p>
<center><hr width=80%></center>

<a name="text"></a>
<h3>How do I effectively use text string search boxes?</h3>
<ul>
  <li>Text based searching generally searches for both names and synonyms carrying the specified text string.</li>
  <li>Text based searches are <b>NOT</b> case sensitive. You can use uppercase, lowercase, or any combination of the two.</li>
  <li>Operators offer the opportunity to narrow or expand a text string search.
    <ul><li>Often the default operator is <b>"contains"</b>, meaning that items beginning, equaling and containing your term appear on the results page. This operator can be
      changed to <b>"begins"</b> or <b>"equals"</b>, where appropriate.</li>
  </ul></li>
</ul>
<p>
<center><hr width=80%></center>

<h3><a name="pick"></a>How do I effectively use pick list searching?</h3>
<ul>
  <li>Pick lists are often used to search fields using controlled vocabularies.</li>
  <li>If a pick list is presented in a box multiple selections can be made by doing the following:</li>
  <ul>
    <li>On a Macintosh, hold the command (Apple) key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items in
    between.</li>
    <li>On a PC, hold the control key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items in between.</li>
  </ul>
</ul>
<p>
<center><hr width=80%></center>

<h3><a name="path"></a>How do I access Pathology images in the database?</h3>
MTB offers many pathology images for viewing. You can view these images by :
<ul>
  <li>conducting a Pathology Images Search and viewing thumbnail images on the Pathology Image Search Results page (clicking on a thumbnail image will display the Pathology Image Detail
  page for that particular image record), or</li>
  <li>clicking on the Pathology Reports/Images link in the Additional Information column on the Tumor Summary Page.</li>
  <li>restricting a Tumor or Advanced search to only those records with associated pathology images</li>
</ul>

<a href="#top">Back to Top</a>

<hr>
<a NAME="searchbox"></a>
<h2>"Search for" Search Box</h2>

The "Search for" Search Box provides a quick keyword search against Tumor, Organ, Strain or Genetics records. For more detailed search strategies, use the Search form specific for the
area of interest.

<hr width=50%>

To use the "Search for" Search Box chose an item or items from the pick list, type your search terms in the text box, and click on the <b>"Go"</b> button.
<ul><li>It is possible to choose multiple options from the pick list, or to search all sections of the database by selecting the option <b>"All Sections"</b>.</li>
  <ul>
    <li>On a Macintosh, hold the command (Apple) key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items in
    between.</li>
    <li>On a PC, hold the control key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items in between.</li>
  </ul>
  <li>Searches will return up to 10 records for each section of the database searched upon. Users may need to scroll down the page to view the results from other sections. At the
  base of each section is a link that will allow you to view <b>ALL</b> records found in that section.</li>
  <li>Results within each section will be arranged alphanumerically by a default sort parameter.</li>
  <li>The total number of records matching your search criteria will be indicated in the Search Summary under the titlebar for each subsection.</li>
  <li>Search box results use the same result display format as the related query forms. Information included in these pages is summarized in the table below.</li>
</ul>

<hr width=50%>

<h3><A name="more"></a>Where can I find more details on each of the searches?</h3>
<table cellpadding=4 border=1 width=100%>
  <tr>
    <th align=left valign=top>If you select...</th>
    <th align=left valign=top>The search returns...</th>
    <th>For more details, see...</th>
  </tr>
  <tr>
    <th align=left valign=top>All sections</th>
    <td align=left valign=top>All sections as described below.</td>
    <TD>&nbsp;</TD>
  </tr>
  <tr>
    <th align=left valign=top>Tumor</th>
    <td valign=top>Tumor records associated with specified classification (e.g. adenoma, carcinoma, etc.) Results also include Tumor name, Organ affected, Treatment type
    and agents, Strain name, Tumor frequency ranges subdivided by sex, Metastases, Links to tumor details pages.</td>
    <td valign=top><a href="#tumorresults">Interpreting the Tumor Search Results page</A></td>
  </tr>
  <tr>
    <th align=left valign=top>Organ</th>
    <td align=left valign=top>Tumor records associated with a specified organ or tissue of origin. Results are formatted the same way as with the Tumor search.</td>
    <td align=left valign=top> <a href="#tumorresults">Interpreting the Tumor Search Results page</A><br><br></td>
  </tr>
  <tr>
    <th align=left valign=top>Strain</th>
    <td align=left valign=top>Strain records associated with a specified strain background. Includes Strain Name, Strain Type, Strain Description Note and links to Strain Details page.</td>
    <td align=left, valign=top><a href="#strainresults">Interpreting the Strain Search Results page</a></td>
  </tr>
  <tr>
    <th align=left valign=top NOWRAP>Gene Symbols/Names&nbsp;</th>
    <td align=left valign=top>Gene symbol, chromosome, and allele pairs associated with strain genotypes and genetic changes observed in tumors.</td>
    <td align=left valign=top><a href="#geneticsresults">Interpreting the Genetics Search results page.</a></td>
  </tr>
</table>
<br>
<a href="#top">Back to Top</A>

<hr>

<h2><a NAME="quickorgan"></a>Quick Organ/Tissue Search</h2>
The Quick Organ/Tissue Search allows you to search for all tumors involving a specific organ of origin. (For more detailed search options, try the
Tumor, Pathology Images, or Advanced Search Forms.)

<hr width=50%>

To use the Quick Organ/Tissue Search chose an item or items from the pick list and click on the <b>"Search"</b> button.
<ul><li>It is possible to choose multiple options from the organ list.</li>
  <ul>
    <li>On a Macintosh, hold the command (Apple) key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items
    in between.</li>
    <li>On a PC, hold the control key down and click the desired items in the list, or hold the shift key down and click two items to select those and all items in between.</li>
  </ul>
  <li>The search will return all the records that match the chosen terms and display them on the Tumor Search Results page.</li>
  <li>Results will be sorted alphabetically by organ of origin.</li>
</ul>
<br><a href="#top">Back to Top</a>

<a NAME="tumor"></a>
<hr>


<h2>Tumor Search Form</h2>
The Tumor Search Form allows users to search for records that match particular tumor-related criteria including organ/tissue of tumor origin, tumor classification, treatment type,
and agents. Searches may be further refined by restricting the found set to only tumors that have been reported to metastasize and/or only tumors for which we have images in
the database.

<hr width=50%>

To use the Tumor Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Organ or Strain of tumor origin. The default is set to <b>"Organ of tumor origin"</b>.</li>
  <li>The number of records returned may be limited to 25, 100, or 500, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"25"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Tumor Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.

<p>
<table BORDER=1>
  <tr>
    <th BGCOLOR="#DDDDDD">Field</th>
    <th BGCOLOR="#DDDDDD">Description</th>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Organ/Tissue of Origin&nbsp;</b></td>
    <td>The specific organ (or tissue or cell type) from which the tumor originated. This may be different than the organ affected, particularly in records for diseases
      that are disseminated or in records for metastatic lesions. The value for this field is selected from a list of controlled vocabulary terms. Multiple terms may be selected
    for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Classification</b></td>
    <td>Standardized names for tumor classifications. This controlled vocabulary was developed using the animal pathology community tumor classification standards
    whenever possible. Multiple terms may be selected for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Name</b></td>
    <td>This field allows for text searching for part of a tumor designation. The user may try this option if the desired organ name or tumor classification is not readily located
    from the provided controlled vocabularies for those data types.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Type</b></td>
    <td>This field allows the user to search for spontaneous tumors or tumors which arose in mice given a particular type of treatment. The default is <b>"Any"</b>.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Agent</b></td>
    <td>This field allows the user to search for tumors which arose in mice treated with a specific agent. Agents may be chemicals, drugs, types of radiation, etc.</td>
  </tr>
  <tr>
    <td valign="top"><b>Metastasis</b></td>
    <td>These options may be used to restrict the search to return only tumors that have been reported to metastasize, and, if desired, those that have been reported to metastasize
    to a particular organ. Multiple organs may be chosen from the list.</td>
  </tr>
  <tr>
    <td valign="top"><b>Pathology Images</b></td>
    <td>This option may be used to restrict a search to only those tumor records with an associated pathology image(s) in the database.</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Queries for Tumors</h3></center>

<hr width=50%>

<h3>Query for tumors that originate in a specific organ/tissue</h3>
Example: Are there reports of mice with pheochromocytomas of the adrenal gland?

<ol>
  <li>Go to the <b>Tumor Search Form</b>.</li>
  <li>Select <b>"Adrenal gland"</b> from the Organ/Tissue of Origin menu.</li>
  <li>Select <b>"pheochromocytoma"</b> from the Tumor Classification menu.</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<hr width=50%>

<h3>Query about tumor metastasis</h3>

Example: Are there reports of spontaneous tumors that metastasize to the liver?

<ol>
  <li>Go to the <b>Tumor Search Form</b>.</li>
  <li>Select <b>"None (spontaneous)"</b> from the Treatment Type menu.</li>
  <li>Check the box beside <b>"Restrict search to metastatic tumors only"</b> under "Metastasis".</li>
  <li>Choose <b>"Liver"</b> from the menu labeled "Metastasizes to the:".</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<p>
<a href="#top">Back to Top</a>

<hr><a NAME="strain"></a><h2>Strain Search Form</h2>

<p>The Strain Search Form allows users to search for records that match particular strain-related criteria including strain name, strain type, and strain genetics. Searches may be
further refined by restricting the found set to only those strain available from The Jackson Laboratory or available from the Mouse Models of Human Cancers Consortium Repository.

<hr width=50%>

To use the Strain Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Strain Name or Strain Type. The default is set to <b>"Strain Name"</b>.</li>
  <li>The number of records returned may be limited to 25, 100, or 500, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"25"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Strain Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.
<br>
<br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Strain name</b></td>
    <td>The name of the strain of mice. The default operator is <b>"contains"</b>. Users may change the operator to <b>"begins with"</b> or <b>"equals"</b>.
      <br><br><small><b><u>NOTE:</u></b> Strain names in MTB include background and genetics information and follow international nomenclature guidelines for naming laboratory strains,
    when possible.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Strain type</b></td>
    <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). The value for this field is selected from a controlled vocabulary.</td>
  </tr>
  <tr>
    <td valign="top"><b>Genetics</b></td>
    <td>Text search by mutant Gene or Allele carried in a strain. This search is done against gene symbol, name, and synonym.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Other Database links&nbsp;</b></td>
    <td>These options allow users to restrict their searches to mice offered by either JAX&reg; Mice or by the NCI Mouse Repository.
      Users may additionally query for a specific strain of mice offered through <a href="http://jaxmice.jax.org/" target="_blank">The Jackson Laboratory's JAX Research Systems</a>,
    by the stock number of that strain.</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Queries for Mouse Strains</h3></center>

<hr width=50%>

<h3>Query for a specific strain of mice</h3>

Example: Find the strain record for BALB/cByJ inbred mice.

<ol>
  <li>Go to the <b>Strain Search Form</b>.</li>
  <li>Type <b>"BALB/cByJ"</b> in the Strain Name text box.</li>
  <li>Choose <b>"inbred"</b> from the Strain Type menu.</li>
  <li>Click on the <b>"Search"</b> button.</li>
  <li>Click on the <b>"BALB/cByJ"</b> hyperlink to view the Strain Tumor Overview for that strain.</li>
</ol>

<p>

<hr width=50%>

<h3>Query by strain type</h3>

Example: Find all the strains in the database that are targeted mutant (knockout) mice where the mutation carried by the mice involves the <i>Kras</i> gene.

<ol>
  <li>Go to the <b>Strain Search Form</b>.</li>
  <li>Select <b>"targeted mutation (knockout)"</b> from the Strain Type value list.</li>
  <li>Type <b>"Kras"</b> in the Gene or Allele text box.</li>
  <li>Click on the <b>"Search"</b> button.</li>
  <li>If the user feels they are seeing too many or too few records they can redo the search and change the selection for <b>"Max number of items returned"</b> to suit
  their particular needs.</li>
</ol>

<p><a href="#top">Back to Top</a>

<hr>
<a name="geneexpression"></a>

<h2>Gene Expression Data Set Search</h2>

Search for cancer-related mouse gene expression data sets that are in NCBI's Gene Expression Omnibus (GEO) archives. 
Searches can be done by organ, tumor classification, mouse strain or expression assay platform to find relevant gene expression data sets. 
Search results provide links to external gene expression data as well as MTB tumor frequency records and strain details. 

<hr width=50%>

If a search returns zero results try limiting the search cirteria to a single category, ie. organ as opposed to organ and platform.

<hr width="50%">
 
The search results will list series containing samples matching the search criteria.
For each series all matching samples will be listed with links to the data sets and MTB tumor frequency records.
Any matching samples that don't belong to a specific series will be listed at the end of the results.



<p>
<a href="#top">Back to Top</a>



<hr>
<a name="genetics"></a>

<h2>Genetics Search Form</h2>

The Genetics Search Form allows users to search for records that match particular genetic-related criteria including genetic marker, allele, and type of mutation in
strain and tumor records.

<hr width=50%>

To use the Genetics Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Gene Symbol, Mutation type, or Chromosome. The default is set to <b>"Gene Symbol"</b>.</li>
  <li>The number of records returned may be limited to 25, 100, or 500, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"100"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Strain and Tumor Genetics Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Gene/Marker</b></td>
    <td>The approved symbol and name for a genetic marker. Searches using this field query gene names, symbols, and synonyms using either the <b>"contains"</b>, <b>"begins with"</b>, or
      <b>"equals"</b> operator.
      <br><br><small><b><u>NOTE:</u></b> If a particular gene is not initially found in MTB, please check the <a href="http://www.informatics.jax.org/javawi2/servlet/WIFetch?page=markerQF" target="_blank">Mouse
    Genome Database</a> to ensure that the gene symbol and/or name being used in the search is current.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Allele</b></td>
    <td>The approved symbol and name for an allele. Searches using this field query allele names, symbols, and synonyms using either the <b>"contains"</b>, <b>"begins with"</b>, or
    <b>"equals"</b> operator.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Chromosome&nbsp;</b></td>
    <td>Mouse chromosome number. The value for this field is selected from a controlled vocabulary. This field allows users to limit their search to genes/loci on a particular
    chromosome.&nbsp;<br><br><small><b><u>NOTE:</u></b> This field is generally empty for transgenes, unless the integration site has been mapped.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Mutations / Abberrations</b></td>
    <td>Users may search for specific genetic changes by choosing one or more of the options within the Mutations/Aberrations list. Users may search for these changes in strains, tumors
    or both by using the radio buttons at the top of the Type of Genetic Change section.</td>
  </tr>
  <tr>
    <td valign="top"><b>Assay Images</b></td>
    <td>When checked the results will only include records with assay images</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Queries for Genetics</h3></center>

<hr width=50%>

<h3>Query by gene name or symbol</h3>
Example: Are there any records in MTB for tumors or strains in which the Kras gene has been examined?
<ol>
  <li>Go to the <b>Genetics Search Form</b>.</li>
  <li>Type <b>"Kras"</b> in the Gene/Marker text box.</li>
  <li>Click on the <b>"Search"</b> button.
    <small><br>
      <br><b><u>NOTE:</u></b> If a particular gene is not initially found in MTB, please check the <a href="http://www.informatics.jax.org/javawi2/servlet/WIFetch?page=markerQF" target="_blank">Mouse
  Genome Database</a> to ensure that the gene symbol and/or name being used in the search is current.</small></li>
</ol>

<hr width=50%>

<h3>Query for a specific genetic change</h3>
Example: Are there records in MTB of tumors associated with genomic instability?
<ol>
  <li>Go to the <b>Genetics Search Form</b>.</li>
  <li>Select <b>"Amplification"</b> from the list of terms for the "Mutations / Aberrations:" </li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<p>
<a href="#top">Back to Top</a>


<hr>
<a name="humangene"></a>

<h2>Human Gene Search Form</h2>

Allows human genes to be used to find orthologous mouse genes and any tumor or strain reference to these genes in MTB.

<hr width=50%>

To use the Human Gene Search form enter a single gene or delimited list of human gene symbols or Entrez Gene ID's and click on the <b>"Search"</b> button.<br>
If you know only part of the gene symbol or want to search for a number of similar symbols use the <b>"Contains"</b>, <b>"Begins with"</b>, or
<b>"Ends with"</b> operator. Entrez Gene ID's can only be used with the "Equals" operator.

<hr width=50%>

<center><h3>Example Queries for Human Genes</h3></center>

<hr width=50%>

<h3>Single query by gene symbol</h3>
Example: Is there a mouse ortholog to the human gene Kras and does MTB contain any records for it?
<ol>
  <li>Go to the <b>Human Gene Search Form</b>.</li>
  <li>Type <b>"Kras"</b> in the Human Symbol text box.</li>
  <li>Click on the <b>"Search"</b> button.
  <br>
</ol>

<hr width=50%>

<h3>Search for any mouse orthologs for the human genes that start with NTRK or EPHA</h3>
<ol>
  <li>Go to the <b>Human Gene Search Form</b>.</li>
  <li>Select <b>"Begins With"</b> from the list of comparison operators.</li>
  <li>Type <b>"NTRK,EPHA"</b> into the Human Sybmol text box.(a number of delimiters are accpeted ie newline, tab, semicolon etc.)</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<p>
<a href="#top">Back to Top</a>


<hr><a NAME="pathology"></a>


<h2>Pathology Images Search Form</h2>

The Pathology Images Search Form allows users to search for records that match particular pathology image-related criteria including organ or tissue of tumor origin, tumor classification, 
diagnosis, stain/method, and antibody.

<hr width=50%>

To use the Pathology Images Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Organ of origin, Strain of origin, Method, or Antibody. The default is set to <b>"Organ of origin"</b>.</li>
  <li>The number of records returned may be limited to 5, 10, or 100, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"5"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Pathology Image Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Organ/Tissue of Origin</b></td>
    <td>The specific organ (or tissue or cell type) from which the tumor originated. This may be different than the organ affected, particularly in records for diseases
      that are disseminated or in records for metastatic lesions. The value for this field is selected from a list of controlled vocabulary terms. Multiple terms may be selected
    for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Classification</b></td>
    <td>Standardized names for tumor classifications. This controlled vocabulary was developed using the animal pathology community tumor classification standards
    whenever possible. Multiple terms may be selected for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Organ/Tissue Affected</b></td>
    <td>The organ in which the lesion was found. The value for this field is selected from a controlled vocabulary. The default is set to <b>"ANY"</b>.</td>
  </tr>
  <tr>
    <td valign="top"><b>Stain/Method</b></td>
    <td>Stain/method of slide preparation. The value for this field is selected from a list of staining methods.</td>
  </tr>
  <tr>
    <td valign="top"><b>Antibody</b></td>
    <td>Specific antibody used in immunohistochemistry (IHC). The value for this field is selected from a list of antibodies used for IHC.</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Queries for Pathology Images</h3></center>

<hr width=50%>

<h3>Query for pathology images by organ of origin</h3>

Example: Retrieve pathology images for tumors arising from the cortex of the adrenal gland.
<ol>
  <li>Go to the <b>Pathology Image Search Form</b>.</li>
  <li>Choose <b>"Adrenal gland - Cortex"</b> from the Organ/Tissue of Origin menu.</li>
  <li>Click on the <b>"Search"</b> button.</li>
  <li>Users may view a Pathology Image Detail page for a record by clicking on the corresponding thumbnail image.
</ol>

<hr width=50%>

<h3>Query for images of tumor samples processed by immunohistochemistry using a particular antibody</h3>

Example: Retrieve records of pathology images in which anti-fillagrin (PRB-417P) was used as a probe.

<ol>
  <li>Go to the <b>Pathology Image Search Form</b>.</li>
  <li>Choose <b>"anti-filaggrin (PRB-417P)"</b> from the Antibody menu.</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<p>
<a href="#top">Back to Top</a>

<hr><a NAME="reference">

<h2>Reference Search Form</h2>

The Reference Search Form allows users to search for records that match particular citation-related criteria from a published or unpublished report including authors,
journal, year, volume, page, or title.

<hr width=50%>

To use the Reference Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Publication Year or First Author. The default is set to <b>"Year"</b>.</li>
  <li>The number of records returned may be limited to 25, 100, or 500, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"25"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Reference Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>First Author&nbsp;</b></td>
    <td>The last name and initials of the first author for a published article. The default operator for this field is <b>"Contains"</b>. <b>"Begins"</b> or <b>"Equals"</b> may also be used.</td>
  </tr>
  <tr>
    <td valign="top"><b>Authors</b></td>
    <td>The last name and initials of any of the authors for a published article. The operator for this field is <b>"Contains"</b>.</td>
  </tr>
  <tr>
    <td valign="top"><b>Journal</b></td>
    <td>The name of the journal in which the published reference appeared. The default operator for this field is <b>"Contains"</b>. <b>"Begins"</b> or <b>"Equals"</b> may also be used.</td>
  </tr>
  <tr>
    <td valign="top"><b>Year</b></td>
    <td>The year a reference was published. The default operator for this field is <b>"Equals"</b>. <b>"Greater Than"</b> or <b>"Less Than"</b> may also be used.</td>
  </tr>
  <tr>
    <td valign="top"><b>Volume</b></td>
    <td>The volume of the journal in which an article was published.</td>
  </tr>
  <tr>
    <td valign="top"><b>Page</b></td>
    <td>The first page number of a published article.</td>
  </tr>
  <tr>
    <td valign="top"><b>Title</b></td>
    <td>Text search for a word or words in the title of an article. The default operator for this field is <b>"Contains"</b>. <b>"Begins"</b> may also be used.</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Queries for References</h3></center>

<hr width=50%>

<h3>Query for references by first author</h3>

Example: Retrieve references where Tyler Jacks is the first author.
<ol>
  <li>Go to the <b>Reference Search Form</b>.</li>
  <li>Type <b>"Jacks T"</b> in the First Author text box.</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<hr width=50%>

<h3>Query for references by year and a term in the article title</h3>

Example: Retrieve references published in 2005 where "retinoblastoma" appears in the title of the article.

<ol>
  <li>Go to the <b>Reference Search Form</b>.</li>
  <li>Type <b>"2005"</b> in the field labeled "Year".</li>
  <li>Type <b>"retinoblastoma"</b> in the field labeled "Title".</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<p>
<a href="#top">Back to Top</a>

<hr>

<a NAME="advanced">

<h2>Advanced Search Form</h2>

The Advanced Search Form allows users to perform complex queries using combinations of strain, tumor, genetics, and pathology image search criteria.

<hr width=50%>

To use the Advanced Search Form chose or enter an item or items from the available search options and click on the <b>"Search"</b> button.

<ul>
  <li>Results are sorted alphanumerically by Tumor Classification, Organ of tumor origin, Strain name, Strain Type, or Treatment Type. The default is set to <b>"Tumor Classification"</b>.</li>
  <li>The number of records returned may be limited to 25, 100, or 500, or users may choose to view all records by choosing the <b>"No Limit"</b> option. The default is set
  to <b>"25"</b> records.</li>
</ul>

<hr width=50%>

The following table describes the fields searchable from the Advanced Search Form. For general search help, please refer to <a href="#queries">How to query the Mouse Tumor
Biology Database</a> elsewhere in this document.<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Strain name</b></td>
    <td>The name of the strain of mice. The default operator is <b>"contains"</b>. Users may change the operator to <b>"begins with"</b> or <b>"equals"</b>.
      <br><br><small><b><u>NOTE:</u></b> Strain names in MTB include background and genetics information and follow international nomenclature guidelines for naming laboratory strains,
    when possible.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Strain type</b></td>
    <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). The value for this field is selected from a controlled vocabulary.</td>
  </tr>
  <tr>
    <td valign="top"><b>Gene or Allele</b></td>
    <td>Text search by mutant Gene or Allele carried in a strain. This search is done against gene symbol, name, and synonym.</td>
  </tr>
  
  <tr>
    <td valign="top" NOWRAP><b>Organ/Tissue of Origin&nbsp;</b></td>
    <td>The specific organ (or tissue or cell type) from which the tumor originated. This may be different than the organ affected, particularly in records for diseases
      that are disseminated or in records for metastatic lesions. The value for this field is selected from a list of controlled vocabulary terms. Multiple terms may be selected
    for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Classification</b></td>
    <td>Standardized names for tumor classifications. This controlled vocabulary was developed using the animal pathology community tumor classification standards
    whenever possible. Multiple terms may be selected for searching, in which case the database will return all records matching at least one of the selected terms.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Type</b></td>
    <td>This field allows the user to search for spontaneous tumors or tumors which arose in mice given a particular type of treatment. The default is <b>"Any"</b>.</td>
  </tr>
  <tr>
    <td valign="top"><b>Metastasis</b></td>
    <td>This option may be used to restrict the search to return only tumors that have been reported to metastasize.</td>
  </tr>
  <tr>
    <td valign="top"><b>Pathology Images</b></td>
    <td>This option may be used to restrict a search to only those tumor records with an associated pathology image(s) in the database.</td>
  </tr>
</table>

<hr width=50%>

<center><h3>Example Advanced Queries</h3></center>

<hr width=50%>

<h3>Query for tumors based on strain type, tumor characteristics and restrict the results to records with pathology images</h3>

Example: Retrieve tumor records for spontaneous mammary gland adenocarcinomas in transgenic mice and only show those with associated pathology images.
<ol>
  <li>Go to the <b>Advanced Search Form</b>.</li>
  <li>Select <b>"transgenic"</b> from the Strain Type menu.</li>
  <li>Select <b>"Mammary gland"</b> from the Organ/Tissue of Origin menu.</li>
  <li>Select <b>"adenocarcinoma"</b> from the Tumor Classification menu.</li>
  <li>Select <b>"None (spontaneous)"</b> from the Treatment Type menu.</li>
  <li>Click on the box beside <b>"Restrict search to entries with associated pathology images"</b>.</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<hr width=50%>

<h3>Query for metastatic tumors from inbred mice</h3>

Example: Retrieve tumor records for spontaneous adenocarcinomas in inbred mice that have been reported to metastasize.
<ol>
  <li>Go to the <b>Advanced Search Form</b>.</li>
  <li>Select <b>"inbred"</b> from the Strain Type menu.</li>
  <li>Select <b>"adenocarcinoma"</b> from the Tumor Classification menu.</li>
  <li>Select <b>"None (spontaneous)"</b> from the Treatment Type menu.</li>
  <li>Click on the box beside <b>"Restrict search to metastatic tumors only"</b>.</li>
  <li>Click on the <b>"Search"</b> button.</li>
</ol>

<hr>
<center><h1> How to Interpret Search Results in MTB</h1></center>
<p><a NAME="interpreting"></a>
<h2>General Tips on Interpreting Search Results</h2>
The following is a brief introduction to the Search Results pages. More detailed descriptions for each page and how to access them are given below.
<ul>
  <li>Data in MTB are presented at increasing levels of granularity on Search Results, Summary, Overview and Detail pages for tumor, strain, genetics, pathology, and reference data.
  The Search Results page provides a general overview of records found by your search, while the other pages provide more in depth information on a record(s).</li>
  <li>Search results are alphanumerically sorted by parameter of choice.</li>
  <li>Each page has a title bar at the top, indicating which level of detail is included on the page. Listed just below this title bar is an indicator of the search parameters, how the records were sorted
    (e.g. organ or tumor name), the display limit of the search, and a text string indicating the number of items displayed vs the number of matching records found by the search. (For
  example, <b>"25 of 50 matching items displayed"</b>.)</li>
  <li>The default for number of records returned by a search is generally 25 (5 for pathology images searches). You can expand the number of records to view, by requesting a larger number of records
  in the search form. The option <b>"No Limit"</b> enables you to view all records.</li>
</ul>

<p><a href="#top">Back to Top</a>

<hr>

<h2><a name="tumorresults"></a>Tumor Search Results</h2>
The Tumor Search Results page presents all tumor records found matching the requested search criteria. Users may refine the search by going back to the form and adding more selections
to the search criteria.

<hr width=50%>

The Tumor Search Results page may be accessed from:
<ul>
  <li>the results page generated by the "Search for" box in the menu bar when searching on either <b>"Tumor"</b> or <b>"Organ"</b></li>
  <li>the Quick Organ/Tissue Search Form</li>
  <li>the Tumor Search Form</li>
  <li>the Advanced Search Form</li>
  <li>the <b>"Tumor Records"</b> hyperlink on the Reference Detail page</li>
  <li>the Number of Associated Tumor Frequencies hyperlinks on the Genetics Change Summary page</li>
</ul>

<hr width=50%>

A Search Summary is located just beneath the heading on the Tumor Search Results page. This summary provides the following information:
<ul>
  <li>a summary of the search parameters and operators chosen for the search. This section is dynamically generated, so it does not show parameters not included in the current search criteria</li>
  <li>the selected sort order, either by Organ of tumor origin or Strain of tumor origin. The default is set to <b>"Organ of tumor origin"</b>.</li>
  <li>the selected display limit, either 25, 100, 500, or No Limit. The default is set to <b>"25"</b></li>
  <li>the number of records displayed as well as the total number of matching records found by the query</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Tumor Search Results page.

<br><br>
<table border="1" valign="top" width="100%">
  <tr>
    <th BGCOLOR="#DDDDDD">Field</th>
    <th BGCOLOR="#DDDDDD">Description</th>
  </tr>
  <tr>
    <th align=left valign=top>Tumor Name<br/></th>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>
  <tr>
    <th align=left valign=top>Organ Affected</th>
    <td>The organ in which the tumor was found. This may be different than the organ of origin, particularly in records for diseases
    that are disseminated or in records for metastatic lesions.</td>
  </tr>
  <tr>
    <th align=left valign=top NOWRAP>Treatment Type (Agents)</th>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment and, if treated, lists the specific agent(s) given.</td>
  </tr>
  <tr>
    <th align=left valign=top>Strain Name</th>
    <td>The name of the strain of mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
  </tr>
  <tr>
    <th align=left valign=top>Strain Types</th>
    <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). One or more types are assigned to each strain.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Tumor Frequency Range&nbsp;</b></td>
    <td valign="top">Tumor Frequency is the percentage of a population of mice that develop a specific lesion.<br>
      <br>
      This field summarizes the the reported frequency of a particular lesion in a specific strain.<br>
      <br>
      The Tumor Frequency Range is often a summary of frequencies reported in different references. Tumor frequency ranges are displayed by
      sex. Separate Tumor Frequency Ranges are displayed for female, male, mixed population, or unspecified sex mice.
      <br><br><small><b><u>NOTE:</u></b> Observations of <b>"zero percent frequency"</b> are recorded in the Mouse Tumor Biology Database.</small><br><br>
      Often, subjective descriptions of tumor frequency are reported rather than exact proportions or percentages. We have assigned non-exact
      text descriptions of tumor frequency the following numerical values for the purpose of database searching and sorting:
      <center><h3>Numerical equivalencies assigned to text descriptions of tumor frequency</h3></center>
      <center>
        <table BORDER=1>
          <tr>
            <th BGCOLOR="#DDDDDD">Frequency Description&nbsp;</th>
            <th BGCOLOR="#DDDDDD">Numerical Equivalent&nbsp;</th>
          </tr>
          <tr>
            <td>very high</td>
            <td>&gt; 80%</td>
          </tr>
          <tr>
            <td>high</td>
            <td>&gt; 50%</td>
          </tr>
          <tr>
            <td>moderate</td>
            <td>&gt; 30%</td>
          </tr>
          <tr>
            <td>low</td>
            <td>&lt; 20%</td>
          </tr>
          <tr>
            <td>very low</td>
            <td>&lt;10%</td>
          </tr>
          <tr>
            <td>sporadic</td>
            <td>&lt; 1%</td>
          </tr>
          <tr>
            <td>observed</td>
            <td>&gt; 0.1%</td>
          </tr>
      </table></center>
    </td>
  </tr>
  <tr>
    <th align="left" valign="top">Metastasizes To</th>
    <td>A list of organs a tumor was found to metastasize to or found to <i>not metastasize to</i>. The negative results are preceded by the word "not" and are listed in italics.
    If no metastases results were reported, this field is left empty.</td>
  </tr>
  <tr>
    <th align=left valign=top>Images</th>
    <td>A camera icon in this column indicates that there are pathology images in MTB associated with one or more of the tumor frequency records represented by that row.</td>
  </tr>
  <tr>
    <th align="left" valign="top">Tumor Summary</th>
    <td>A hyperlink to more specific information for tumor frequency records sharing the same tumor name, organ of origin/organ affected, strain, treatment,
    and metastatic status as displayed in the corresponding row.</td>
  </tr>
</table>

<br>
<a href="#top">Back to Top</A>
<hr>


<a name="tumorsummary"></a><h2>Tumor Summary</h2>
The Tumor Summary page presents a summary of tumors arising in the same strain under the same conditions and may include tumor frequency records from multiple citations.<br>

<hr width=50%>

The Tumor Summary page may be accessed by clicking on:
<ul>
  <li>the Tumor Summary links on the Tumor Search Results Page, or</li>
  <li>the numerical links in the Number of Tumor Group Records column on the Strain Overview pages.</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Tumor Summary page.<br>
<br>
<small><b><u>NOTE:</u></b> Some of the fields presented on the top portion of the Tumor Summary page are populated dynamically and are only visible if that field contains data, otherwise the corresponding
area is hidden from view.</small>
<br>
  
  <table BORDER=1>
  <tr>
    <th BGCOLOR="#DDDDDD">Field</th>
    <th BGCOLOR="#DDDDDD">Description</th>
  </tr>
  <tr>
<td valign="top"><b>Tumor Name</br></b></td>
<td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
</tr>
<tr>
  <td valign="top"><b>Treatment Type<br></b></td>
  <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment.</td>
</tr>
<tr>
  <td valign="top"><b>Tumor synonym(s)</b></td>
  <td>Alternate names used to refer to this lesion.</td>
</tr>
<tr>
  <td valign="top"><b>Strain</b></td>
  <td>The name of the strain of mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
</tr>
<tr>
  <td valign="top"><b>Strain Type(s)</b></td>
  <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). One or more types are assigned to each strain.</td>
</tr>
<tr>
  <td valign="top"><b>General Note</b></td>
  <td>This is a textual description of the strain, often including details of the generation of the strain.</td>
</tr>
<tr>
  <td valign="top"><b>Strain Synonym</b></td>
  <td>Alternate names used to refer to this strain.</td>
</tr>
<tr>
  <td valign="top"><b>MTB ID</b></td>
  <td>This is a identification number given to each unique tumor frequency record in the MTB database.</td>
</tr>
<tr>
  <td valign="top"><b>Organ Affected</b></td>
  <td>The organ in which the lesion was found.</td>
</tr>
<tr>
  <td valign="top" NOWRAP><b>Treatment Type/Agent(s)&nbsp;</b></td>
  <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment and, if treated, lists the specific agent(s) given.</td>
</tr>
<tr>
  <td valign="top"><b>Strain Sex</b></td>
  <td>Indicates the sex of the mice, which may be female, male, mixed population, or unspecified sex.</td>
</tr>
<tr>
  <td valign="top"><b>Reproductive Status</b></td>
  <td>Indicates the reported reproductive status of the mice.</td>
</tr>
<tr>
  <td valign="top"><b>Frequency</b></td>
  <td valign="top">Tumor Frequency is the percentage of a population of mice that develop a specific lesion.<br>
    <br>
    This value may be a distinct number, a range, or a qualified comment.
    <br><br><small><b><u>NOTE:</u></b> Observations of <b>"zero percent frequency"</b> are recorded in the Mouse Tumor Biology Database.</small><br><br>
    Often, subjective descriptions of tumor frequency are reported rather than exact proportions or percentages. We have assigned non-exact
    text descriptions of tumor frequency the following numerical values for the purpose of database searching and sorting:
    <center><h3>Numerical equivalencies assigned to text descriptions of tumor frequency</h3></center>
    <center>
      <table BORDER=1>
        <tr>
          <th BGCOLOR="#DDDDDD">Frequency Description&nbsp;</th>
          <th BGCOLOR="#DDDDDD">Numerical Equivalent&nbsp;</th>
        </tr>
        <tr>
          <td>very high</td>
          <td>&gt; 80%</td>
        </tr>
        <tr>
          <td>high</td>
          <td>&gt; 50%</td>
        </tr>
        <tr>
          <td>moderate</td>
          <td>&gt; 30%</td>
        </tr>
        <tr>
          <td>low</td>
          <td>&lt; 20%</td>
        </tr>
        <tr>
          <td>very low</td>
          <td>&lt;10%</td>
        </tr>
        <tr>
          <td>sporadic</td>
          <td>&lt; 1%</td>
        </tr>
        <tr>
          <td>observed</td>
          <td>&gt; 0.1%</td>
        </tr>
    </table></center>
  </td>
</tr>
<tr>
  <td valign="top"><b>Age of Onset</b></td>
  <td>The age of the mice at the time the lesion first arose, often the age when the tumors first are discernible. For many lesions this is not readily apparent as they are not diagnosed until after the mice have died.</td>
</tr>
<tr>
  <td valign="top"><b>Age of Detection</b></td>
  <td>The age of the mice at the time of tumor detection. This is often the age of death.</td>
</tr>
<tr>
  <td valign="top"><b>Additional Information</b></td>
  <td>This column provides text links for Pathology Reports/Images, Genetics, and Additional Notes, where available, for a specific record. Clicking on these hyperlinks
    opens a pop-up window displaying the related data. Pathology information may include textual descriptions as well as images. Genetics information details the somatic mutations observed in the lesion.
  Additional Notes may include treatment details, gene expression levels, and/or other tumor-related information.</td>
</tr>
<tr>
  <td valign="top"><b>Reference</b></td>
  <td>A unique identification number (<b>"J:number"</b>) for the reference for this record. This number is hyperlinked to the Reference Detail page for that particular reference.</td>
</tr>
</table>

<br>
<a href="#top">Back to Top</A>

<hr>


<a name="strainresults"></a><h2>Strain Search Results</h2>

The Strain Search Results page presents all strain records found matching the requested search criteria. Users may refine the search by going back to the form and adding more selections
to the search criteria.

<hr width=50%>

The Strain Search Results page may be accessed from:
<ul>
  <li>the results page generated by the "Search for" box in the menu bar when searching on "Strain"</li>
  <li>the Strain Search Form</li>
  <li>the <b>"Strains"</b> hyperlink on the Reference Detail page</li>
  <li>the Genotype hyperlinks on the Genetics Search Results page</li>
</ul>

<hr width=50%>

A Search Summary is located just beneath the heading on the Strain Search Results page. This summary provides the following information:
<ul>
  <li>a summary of the search parameters and operators chosen for the search. This section is dynamically generated, so it does not show parameters not included in the current search criteria</li>
  <li>the selected sort order, either by Strain Name or Strain Type. The default is set to <b>"Strain Name"</b></li>
  <li>the selected display limit, either 25, 100, 500, or No Limit. The default is set to <b>"25"</b></li>
  <li>the number of records displayed as well as the total number of matching records found by the query</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Strain Search Results page.

<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Strain Name&nbsp;</b></td>
    <td>The name of the strain of mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
  </tr>
  <tr>
    <td valign="top"><b>Strain Type</b></td>
    <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). One or more types are assigned to each strain.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Strain Notes&nbsp;</b></td>
    <td>This is a textual description of the strain, often including details of the generation of the strain.</td>
  </tr>
</table>

<br>
<a href="#top">Back to Top</a>
<hr>

<a name="straindetail"></a><h2>Strain Tumor Overview (Collapsed View and Expanded View)</h2>


The Strain Tumor Overview page presents a summary of the tumor frequency records associated with a particular strain and may include data from multiple citations.

<hr width=50%>

The Strain Tumor Overview page may be accessed by clicking on:
<ul>
  <li>the hyperlinked strain names found on several database pages</li>
  <ul>
    <li>Strain Search Results</li>
    <li>Tumor Search Results</li>
    <li>Tumor Summary</li>
    <li>Pathology Image Search Results</li>
    <li>Pathology Image Details</li>
  </ul>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Strain Tumor Overview page. Most of these are visible on both the Collapsed View and Expanded View pages. Some are visible only on the 
Expanded View pages (as indicated below). Users may switch between the two views by clicking on the appropriate hyperlink in the upper right-hand portion of the page. Tumor listings are grouped by 
Tumor Name, Organ Affected, and Treatment Type on the Collapsed View pages, and they are grouped by Tumor Name, Organ Affected, Treatment Type, Treatment Agent, and Metastatic Status on the Expanded View 
pages.<br>
<br>
<small><b><u>NOTE:</u></b> Some of the fields presented on the Strain Tumor Overview page are populated dynamically and are only visible if that field contains data, otherwise the corresponding
area is hidden from view.</small>
<br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Strain</b></td>
    <td>The name of the strain of mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
  </tr>
  <tr>
    <td valign="top"><b>Strain Synonyms</b></td>
    <td>Alternate names used to refer to this strain.</td>
  </tr>
  <tr>
    <td valign="top"><b>Strain Note</b></td>
    <td>This is a textual description of the strain, often including details of the generation of the strain.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Name</b></td>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>
  <tr>
    <td valign="top"><b>Organ Affected</b></td>
    <td>The organ in which the lesion was found.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Type/<br>Agents (Expanded View)</b></td>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment. Specific treatment agent(s) are given on the Expanded View pages.</td>
  </tr>
  <tr>
    <td valign="top"><b>Metastasizes to<br>(Expanded View)</b></td>
    <td>A list of organs a tumor was found to metastasize to or found to <i>not metastasize to</i>. The negative results are preceded by the word "not" and are listed in italics.
    If no metastases results were reported, this field is left empty.</td>
  </tr>
  <tr>
    <td valign="top"><b>References<br>(Expanded View)</b></td>
    <td>A unique identification number (<b>"J:number"</b>) for the reference for this record. This number is hyperlinked to the Reference Detail page for that particular reference.</td>
  </tr>
  <tr>
    <td valign="top"><b>Number of Tumor Group/Sub-type Records</b></td>
    <td>This indicates the number of tumor frequencies that fit the criteria shown on the corresponding line of the table (unless all the associated frequencies have a value of "0%"
      (see <small><b>NOTE</b></small>)). Clicking on the hyperlinked number will go to the Tumor Summary page for those tumor frequencies.<br>
      <br><small><b><u>NOTE:</u></b> When a "0" is present in this column it does NOT indicate zero associated frequencies, but rather it is an indication that all 
    the associated frequencies have a value of 0%.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Other Database Links</b></td>
    <td>Hyperlinks to additional related data held in external resources:
      <ul>
        <li><a href="http://www.informatics.jax.org/external/festing/search_form.cgi" target="_blank">Festing's List of Inbred Strains</a></li>
        <li><a href="http://www.jax.org/phenome" target="_blank">Mouse Phenome Database (MPD)</a></li>
        <li><a href="http://jaxmice.jax.org/" target="_blank">JAX&reg; Mice</a></li>
        <li><a href="http://ccm.ucdavis.edu/bcancercd/" target="_blank">Mammary Cancer Comparative Pathology Web Site</a></li>
        <li><a href="http://mammary.nih.gov/index.html" target="_blank">Biology of the Mammary Gland Web Site</a></li>
      </ul>
    </td>
  </tr>
</table>

<p>

<a href="#top">Back to Top</a>

<hr>

<a name="geneticsresults"></a><h2>Genetics Search Results</h2>

The Genetics Search Results page presents all records for strain and tumor genetics found matching the requested search criteria. Users may refine the search by going back to the form and
adding more selections to the search criteria.

<hr width=50%>

The Genetics Search Results page may be accessed from:
<ul>
  <li>the results page generated by the "Search for" box in the menu bar when searching on "Genetics"</li>
  <li>the Genetics Search Form</li>
</ul>

<hr width=50%>

A Search Summary is located just beneath the heading on the Genetics Search Results page. This summary provides the following information:
<ul>
  <li>a summary of the search parameters and operators chosen for the search. This section is dynamically generated, so it does not show parameters not included in the current search criteria</li>
  <li>the selected sort order, either by Gene Symbol, Mutation Type, or Chromosome. The default is set to <b>"Gene Symbol"</b></li>
  <li>the selected display limit, either 25, 100, 500, or No Limit. The default is set to <b>"100"</b></li>
  <li>the number of records displayed for strain genetics records and the number of records displayed for genetic changes observed in lesions</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Genetics Search Results page.

<br><br>

<table border=1>
  <tr>
    <td BGCOLOR="#BBBBBB" colspan=2><h3>Genes, Alleles and Transgenes carried by Strains</h3></td>
  </tr>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  
  <tr>
    <td valign="top" NOWRAP><b>Mouse Chromosome&nbsp;</b></td>
    <td>The mouse chromosome of the marker with which the allele is associated. This field is generally empty for transgenes, unless the integration site has been mapped.</td>
  </tr>
  <tr>
    <td valign="top"><b>Symbol, Name</b></td>
    <td>The official gene symbol and gene name for the genetic marker associated with the displayed genotype.<br>
      <br>
      <small><b><u>NOTE:</u></b> Because many transgenes carry genes from organisms other than mouse, the source organism of the expressed
    component of the transgene is indicated in parentheses after the gene name for records of transgenes.</small></td>
  </tr>
  <tr>
    <td valign="top"><b>Type</b></td>
    <td>This indicates the allele type (e.g. normal, chemically induced, targeted mutation, transgene, etc.).</td>
  </tr>
  <tr>
    <td valign="top"><b>Genotype</b></td>
    <td>This indicates the allele pair. The number in parentheses indicates the number of Strain records associated with that allele pair. Clicking on
    the hyperlink will access the Strain Search Results page showing the associated strain records.</td>
  </tr>
  <tr>
    <td BGCOLOR="#BBBBBB" colspan=2><h3>Genetic Markers analyzed in the Tumor</h3></td>
  </tr>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Mouse Chromosome&nbsp;</b></td>
    <td>The mouse chromosome of the genetic change, where applicable.</td>
  </tr>
  <tr>
    <td valign="top"><b>Gene Symbol, Name</b></td>
    <td>The official gene symbol and gene name for the genetic marker associated with the displayed genetic change.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Marker Type Observed&nbsp;</b></td>
    <td>This indicates the type of genetic change observed. The number in parentheses indicates the number of specific genetic changes associated with that genetic change type. Clicking on
    the hyperlink will access the Genetic Change Summary page showing the associated genetic change records.</td>
  </tr>
  
  <tr>
    <td BGCOLOR="#BBBBBB" colspan=2><h3>Cytogenetic observations in the Tumor</h3></td>
  </tr>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Mouse Chromosome&nbsp;</b></td>
    <td>The mouse chromosome of the genetic change, where applicable.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Type&nbsp;</b></td>
    <td>This indicates the type of cytogenetic observation. The number in parentheses indicates the number of cytogenetic observations of that type for the specified chromosomes. Clicking on
    the hyperlink will access the Cytogenetic Summary page showing the associated cytogenetic records.</td>
  </tr>
  
</table>

<p>
<a href="#top">Back to Top</a>
<hr>

<a name="geneticchange"></a><h2>Genetic Change Summary</h2>

The Genetic Change Summary page presents a summary of the tumor frequencies associated with the specific somatic genetic change indicated.

<hr width=50%>

The Genetic Change Summary page may be accessed by clicking on:
<ul>
  <li>the hyperlinked number in the Marker Type Observed column in the Genetic Changes in Tumors section of the Genetic Search Results page, which is located on the lower portion of the page.</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Genetic Change Summary page.

<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Gene Symbol</b></td>
    <td>The official gene symbol for the genetic marker.</td>
  </tr>
  <tr>
    <td valign="top"><b>Gene Name</b></td>
  <td>The official gene name for the genetic marker.</td></tr>
  <tr>
    <td valign="top"><b>Mouse Chromosome</b></td>
    <td>The mouse chromosome of the marker with which the genetic change is associated. This field is generally empty for transgenes, unless the integration site has been mapped.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Genetic Change Type&nbsp;</b></td>
    <td>This indicates the type of genetic change observed.</td>
  </tr>
  <tr>
    <td valign="top"><b>Mutation Location</b></td>
    <td>This field provides the user with a more detailed description of the mutation observed.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Name</b></td>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Type</b></td>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment.</td>
  </tr>
  <tr>
    <td valign="top"><b>Number of Associated Tumor Frequencies</b></td>
    <td>This indicates the number of tumor frequency records associated with that particular genetic change. Clicking on the hyperlinked number will bring the user to a Tumor Search Results
    page summarizing all of the associated tumor frequency records.</td>
  </tr>
</table>


<p>
<a href="#top">Back to Top</a>
<hr>

<a name="cytogeneticsummary"></a><h2>Cytogenetic Summary</h2>

The Cytogenetic Summary page presents a summary of the tumor frequencies associated with the specific cytogenetic change.

<hr width=50%>

The Cytoenetic Summary page may be accessed by clicking on:
<ul>
  <li>the hyperlinked number in the Marker Type Observed column in the Cytogenetic Observations in Tumors section of the Genetic Search Results page, which is located on the lower portion of the page.</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Cytogenetic Summary page.

<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Chromosomes</b></td>
    <td>The mouse chromosomes associated with the cytogenetic observation.</td>
  </tr>
  <tr>
    <td valign="top"><b>Name</b></td>
  <td>The official gene name for the genetic marker.</td></tr>
  <tr>
  <tr>
    <td valign="top"><b>Note</b></td>
  <td>Additional notes releating to the observation.</td></tr>
  <tr>
  <tr>
    <td valign="top" NOWRAP><b>Type&nbsp;</b></td>
    <td>This indicates the type of genetic change observed.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Assay Type&nbsp;</b></td>
    <td>This indicates the type of assay.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Name</b></td>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>
  <tr>
    <td valign="top"><b>Treatment Type</b></td>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Images</b></td>
    <td>Contains any assay images for the cytogenetic record.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Summary</b></td>
    <td>A link to the Tumor Summary for this Tumor Frequency Record.</td>
  </tr>
</table>


<p>
<a href="#top">Back to Top</a>

<hr>

<a name="pathresults"></a><h2>Pathology Image Search Results</h2>

The Pathology Image Search Results page presents all pathology image records found matching the requested search criteria. Thumbnail images serve as hyperlinked to Pathology Image Detail pages.
Users may refine the search by going back to the form and adding more selections to the search criteria.

<hr width=50%>

The Pathology Image Search Results page may be accessed from:
<ul>
  <li>the Pathology Image Search Form</li>
  <li>the <b>"Pathology Images"</b> hyperlink on the Reference Detail page</li>
</ul>

<hr width=50%>

A Search Summary is located just beneath the heading on the Pathology Image Search Results page. This summary provides the following information:
<ul>
  <li>a summary of the search parameters and operators chosen for the search. This section is dynamically generated, so it does not show parameters not included in the current search criteria</li>
  <li>the selected sort order, either by Organ of origin, Strain of origin, Method, or Antibody. The default is set to <b>"Organ of origin"</b></li>
  <li>the selected display limit, either 5, 10, 100, or No Limit. The default is set to <b>"5"</b></li>
  <li>the number of records displayed as well as the total number of matching records found by the query</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Pathology Image Search Results page.

<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>MTB ID</b></td>
    <td>This is a identification number given to each unique tumor frequency record in the MTB database.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor name</b></td>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Organ Affected&nbsp;</b></td>
    <td>The organ in which the lesion was found.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Treatment Type/Agent&nbsp;</b></td>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment and, if treated, lists the specific agent(s) given.</td>
  </tr>
  <tr>
    <td valign="top"><b>Strain name/<br>Strain Sex/<br>Reproductive Status</b></td>
    <td valign="top">The name, sex, and reproductive status of the mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
  </tr>
  <tr>
    <td valign="top"><b>Tumor Frequency</b></td>
    <td valign="top">Tumor Frequency is the percentage of a population of mice that develop a specific lesion.<br>
      <br>
      This value may be a distinct number, a range, or a qualified comment.
      <br><br><small><b><u>NOTE:</u></b> Observations of <b>"zero percent frequency"</b> are recorded in the Mouse Tumor Biology Database.</small><br><br>
      Often, subjective descriptions of tumor frequency are reported rather than exact proportions or percentages. We have assigned non-exact
      text descriptions of tumor frequency the following numerical values for the purpose of database searching and sorting:
      <center><h3>Numerical equivalencies assigned to text descriptions of tumor frequency</h3></center>
      <center>
        <table BORDER=1>
          <tr>
            <th BGCOLOR="#DDDDDD">Frequency Description&nbsp;</th>
            <th BGCOLOR="#DDDDDD">Numerical Equivalent&nbsp;</th>
          </tr>
          <tr>
            <td>very high</td>
            <td>&gt; 80%</td>
          </tr>
          <tr>
            <td>high</td>
            <td>&gt; 50%</td>
          </tr>
          <tr>
            <td>moderate</td>
            <td>&gt; 30%</td>
          </tr>
          <tr>
            <td>low</td>
            <td>&lt; 20%</td>
          </tr>
          <tr>
            <td>very low</td>
            <td>&lt;10%</td>
          </tr>
          <tr>
            <td>sporadic</td>
            <td>&lt; 1%</td>
          </tr>
          <tr>
            <td>observed</td>
            <td>&gt; 0.1%</td>
          </tr>
      </table></center>
    </td>
  </tr>
  <tr>
    <td valign="top"><b>Age at Necropsy</b></td>
    <td>The age of the mouse when the necropsy was performed.</td>
  </tr>
  <tr>
    <td valign="top"><b>Description</b></td>
    <td>The textual pathology description associated with a particular tumor frequency record.</td>
  </tr>
  <tr>
    <td valign="top"><b>Image ID</b></td>
  <td>This is a identification number given to each unique image in the MTB database.</td></tr>
  <tr>	
  <tr>
    <td valign="top"><b>Source of Image</b></td>
    <td>The source of the pathology descriptions and/or images and may be either a published reference or a personal communication (for records submitted directly to MTB).</td>
  </tr>	
  <tr>
    <tr>
    <td valign="top"><b>Pathologist</b></td>
    <td>The name of the pathologist.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Method/Stain</b></td>
    <td>The method/stain used in the preparation of the slides/tissues for the displayed image.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Image Caption</b></td>
    <td>This is a detailed description provided a veterinary pathologist. This additional information helps facilitate more accurate interpretation of pathology images by non-pathologists.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Image Link</b></td>
    <td>Thumbnail images are hyperlinked to the Pathology Image Detail page for that particular image.</td>
  </tr>
</table>

<p>
<a href="#top">Back to Top</a>
<hr>

<a name="pathdetail"></a><h2>Pathology Image Detail</h2>

The Pathology Image Detail page presents a larger version of the selected pathology image as well as detailed pathology information associated with that image.

<hr width=50%>

The Pathology Image Detail page may be accessed by clicking on:
<ul>
  <li>the thumbnail images on the Pathology Image Search Results pages</li>
  <li>the thumbnail images on the Pathology Report/Image pop-up windows</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Pathology Image Detail page.<br>
<br>
<small><b><u>NOTE:</u></b> Some of the fields presented on the Pathology Image Detail page are populated dynamically and are only visible if that field contains data, otherwise
the corresponding area is hidden from view.</small>
<br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Caption</b></td>
    <td>This is a detailed description provided a veterinary pathologist. This additional information helps facilitate more accurate interpretation of pathology images by non-pathologists.</td>
  </tr>
   <tr>
    <td valign="top"><b>Description</b></td>
    <td>A discription of the image.</td>
  </tr>
  <tr>
    <td valign="top"><b>Notes</b></td>
    <td>Any notes included with the image.</td>
  </tr>
  <tr>
    <td valign="top"><b>Contributor</b></td>
    <td>The veterinary pathologist who provided the caption for the displayed image.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Pathologist</b></td>
    <td>The veterinary pathologist credited with displayed image.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Copyright</b></td>
    <td>Any copyright details for the displayed image.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Method</b></td>
    <td>The method used in the preparation of the slides/tissues for the displayed image.</td>
  </tr>
  
  <tr>
    <td valign="top"><b>Image Probes</b></td>
    <td>Details of any probes used in the image</td>
  </tr>
  
  <tr>
    <td valign="top"><b>MTB ID</b></td>
    <td>The id of the the tumor frequency record this image is assoicated with. Links to the tumor summary</td>
  </tr>	
  <tr>
    <td valign="top"><b>Tumor Name</b></td>
    <td>The tumor name is generally a combination of organ of origin name and tumor classification.</td>
  </tr>	
  <tr>
    <td valign="top" NOWRAP><b>Treatment type/Agent&nbsp;</b></td>
    <td>Indicates whether the tumor was spontaneous or arose in mice that had received some form of treatment and, if treated, lists the specific agent(s) given.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Tumor synonyms</b></td>
    <td>Alternate names used to refer to this lesion.</td>
  </tr>
  <tr>
    <td valign="top"><b>Organ Affected</b></td>
    <td>The organ in which the lesion was found.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Frequency</b></td>
    <td valign="top">Tumor Frequency is the percentage of a population of mice that develop a specific lesion.<br>
      <br>
      This value may be a distinct number, a range, or a qualified comment.
      <br><br><small><b><u>NOTE:</u></b> Observations of <b>"zero percent frequency"</b> are recorded in the Mouse Tumor Biology Database.</small><br><br>
      Often, subjective descriptions of tumor frequency are reported rather than exact proportions or percentages. We have assigned non-exact
      text descriptions of tumor frequency the following numerical values for the purpose of database searching and sorting:
      <center><h3>Numerical equivalencies assigned to text descriptions of tumor frequency</h3></center>
      <center>
        <table BORDER=1>
          <tr>
            <th BGCOLOR="#DDDDDD">Frequency Description&nbsp;</th>
            <th BGCOLOR="#DDDDDD">Numerical Equivalent&nbsp;</th>
          </tr>
          <tr>
            <td>very high</td>
            <td>&gt; 80%</td>
          </tr>
          <tr>
            <td>high</td>
            <td>&gt; 50%</td>
          </tr>
          <tr>
            <td>moderate</td>
            <td>&gt; 30%</td>
          </tr>
          <tr>
            <td>low</td>
            <td>&lt; 20%</td>
          </tr>
          <tr>
            <td>very low</td>
            <td>&lt;10%</td>
          </tr>
          <tr>
            <td>sporadic</td>
            <td>&lt; 1%</td>
          </tr>
          <tr>
            <td>observed</td>
            <td>&gt; 0.1%</td>
          </tr>
      </table></center>
    </td>
  </tr>
  <tr>
    <td valign="top"><b>Strain name</b></td>
    <td>The name of the strain of mice. Strain names in MTB follow international nomenclature rules whenever possible.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Strain Type</b></td>
    <td>Method of strain derivation (e.g. inbred strains, targeted mutations, transgenics, etc.). One or more types are assigned to each strain.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Strain Synonyms</b></td>
    <td>Alternate names used to refer to this strain.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Strain Sex</b></td>
    <td>Indicates the sex of the mice, which may be female, male, mixed population, or unspecified sex.</td>
  </tr>	
  <tr>
    <td valign="top"><b>Reproductive Status</b></td>
    <td>Indicates the reported reproductive status of the mice.</td>
  </tr>
</table>	

<p>

<a href="#top">Back to Top</a>

<hr>


<a name="humangeneresults"></a><h2>Human Gene Search Results</h2>

A Search Summary is located just beneath the heading on the Strain Search Results page. 
<br>
This summary provides the following information:
<ul>
  <li>A summary of the search parameters and operators chosen for the search.</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Human Gene Search Results page.

<br><br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Human Gene Symbol&nbsp;</b></td>
    <td>Human Gene Symbol(s) that match the search criteria</td>
  </tr>
  <tr>
    <td valign="top"><b>Mouse Gene Symbol</b></td>
    <td>An orthologous Mouse Gene</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Name</b></td>
    <td>The name of the mouse gene&nbsp;</td>
  </tr>
  
  <tr>
    <td valign="top"><b>Alleles/Transgenes in Strains</b></td>
    <td>The number of Strains in MTB with a reference to this gene</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Classes of Tumor Specific Alterations</b></td>
    <td>The number of tumor records in MTB involving this gene</td>
  </tr>
</table>

<br>
<a href="#top">Back to Top</a>

<hr>

<a name="referenceresults"></a><h2>Reference Search Results</h2>

The Reference Search Results page presents all reference records found matching the requested search criteria. Users may refine the search by going back to the form and adding more selections
to the search criteria.

<hr width=50%>

The Reference Search Results page may be accessed from:
<ul>
  <li>the Reference Search Form</li>
</ul>

<hr width=50%>

A Search Summary is located just beneath the heading on the Reference Search Results page. This summary provides the following information:
<ul>
  <li>a summary of the search parameters and operators chosen for the search. This section is dynamically generated, so it does not show parameters not included in the current search criteria</li>
  <li>the selected sort order, either by Year or First Author. The default is set to <b>"Year"</b></li>
  <li>the selected display limit, either 25, 100, 500, or No Limit. The default is set to <b>"25"</b></li>
  <li>the number of records displayed as well as the total number of matching records found by the query</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Reference Search Results page.

<br><br>
<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Accession ID</b></td>
    <td>A unique identification number (<b>"J:number"</b>) for the reference for this record. This number is hyperlinked to the Reference Detail page for
    that particular reference.</td>
  </tr>
  <tr>
    <td valign="top"><b>Reference</b></td>
    <td>The details for each reference. This includes authors, title, journal, year, volume, issue, and pages for published references, and author, title,
    and year for personal communications.</td>
  </tr>
</table>
<p>
<a href="#top">Back to Top</a><p>
<hr>

<a name="referencedetail"></a><h2>Reference Detail</h2>

The Reference Detail page presents a full citation and abstract for a single reference, as well as a summary of associated information in MTB and a link to the same reference in MGI.

<hr width=50%>

The Reference Detail page may be accessed by clicking on the <b>"J:number"</b> links on:
<ul>
  <li>the Reference Search Results page</li>
  <li>the Tumor Summary page</li>
  <li>the Pathology Reports/Images pop-up page</li>
  <li>the Genetics pop-up page</li>
  <li>the Additional Notes pop-up page</li>
  <li>the Strain Tumor Overview (Expanded View) page</li>
  <li>the Pathology Image Detail page</li>
</ul>

<hr width=50%>

The following table describes the fields displayed on the Reference Detail page.<br>
<br>

<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top"><b>Reference</b></td>
    <td>The full citation for the selected reference. This includes title, authors, journal, volume, issue, year, pages, and abstract for published references, and title,
    author, and year for personal communications.</td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Additional Information in MTB</b></td>
    <td>This section contains a summary of the tumor frequency records, strains, and pathology images in MTB from the selected reference and provides hyperlinks to that data.</td>
  </tr>
  <tr>
    <td valign="top"><b>Other Accession IDs</b></td>
    <td>The unique identification number (<b>"J:number"</b>) for this reference. The accompanying text, "Mouse Genome Informatics", is hyperlinked to the MGI Reference Detail page for
    that particular reference.</td>
  </tr>
</table>

<p>
<a href="#top">Back to Top</a>

<hr>


<hr>


<a href="#top">Back to Top</a>

<hr>
  
  <a name="dynamicgrid"></a><h2>Dynamic Tumor Frequency Grid</h2>

  The dynamic tumor frequency grid presents the same information as the existing tumor frequency grid,
  but allows the data to be refined by the user.<br>
  Individual strain families and organ groups can be selected to generate a customized grid.<br>
  In the resulting grid, individual strains and organs can be selected to further refine the grid's contents. 
  Use the check boxes to select the desired strain and organ rows and columns, then click the 'Generate Grid' button.

<p>
<a href="#top">Back to Top</a>

<hr>
  
  <a name="pdxsearch"></a><h2>PDX Search Form</h2>

 Search for PDX models by primary cancer site, diagnosis, and gene variant.<br><br>
 
 <b>Tumor type</b><br>
    The primary site and diagnosis fields are string searches and are not hierarchical. <br>
    So "non-small cell lung cancer" will not return "lung squamous cell carcinoma."<br>
 
 <br>
 <b>Gene list</b>
 <br>
    The gene lists in the drop down menus are not comprehensive.<br>
    They are a subset of genes that are of interest in cancer (derived from the CTP panel).<br>
    The gene variant field contains any gene for which our assays find variants.<br>
    This field only recognizes the official HGNC gene symbol.<br><br>

 Or select a gene to see expression levels for that gene across PDX models.<br>
 Or select a gene to see models where that gene is amplified or deleted.

<p>
<a href="#top">Back to Top</a>


<hr>
  
  <a name="pdxresults"></a><h2>PDX Search Results</h2>

  PDX models matching the search criteria are displayed in a dynamic table.<br>
  The results can be sorted by any column. Columns can be resized or hidden.<br>
  Click the model ID to go to the model details page to see any additional data.<br>
  To send an email requesting additional information on PDX models select models using the check boxes and click 'Request Details' button.

<p>
<a href="#top">Back to Top</a>


<hr>
  
  <a name="pdxdetails"></a><h2>PDX Details</h2>
  
  Variant, expression and copy number data may not be available for all models.<br>
  
  <a name="pdxVariant"></a><b>Variant data</b>
  <br>

The variant data (point mutations and indels) are analyzed from next-generation sequencing using various capture panels:

<ul>
    <li>Truseq (deprecated) - The Illumina TruSeq Amplicon Cancer Panel covers 48 cancer-related genes. <a href="http://www.illumina.com/products/truseq_amplicon_cancer_panel.html"> Link.</a></li>

    <li>CTP - The JAX Cancer Treatment Profile panel covers 358 cancer-associated genes <a href="https://www.jax.org/clinical-genomics/jax-cancer-treatment-profile"> Link.</a></li>

    <li>Whole Exome (limited number of samples assayed) - Agilent SureSelect human exon capture. </li>
</ul>

The analysis of the sequencing output uses the Xenome tool to remove contaminating mouse sequences before alignment and variant calling.<br>
BWA, GATK, and SnpEff are utilized for alignment (GRCh38 human reference), variant calling and annotation.<br>
The variant results also include clinical annotations using the JAX proprietary clinical knowledge base (CKB).<br>
The high quality variants are further filtered to remove likely false positives,<br>
variants with low and modifier impact based on SnpEff annotations, and putative germline variants.<br>
However, all high quality variants that are listed in the JAX CKB database with associated drug efficacy<br>
or cancer-related evidences are retained regardless of the filters. 

  The following table describes the fields used in the variant summary data.<br>
<br>
<table BORDER=1>
  <tr>
    <td BGCOLOR="#DDDDDD"><b>Field</b></td>
    <td BGCOLOR="#DDDDDD"><b>Description</b></td>
  </tr>
  <tr>
    <td valign="top" NOWRAP><b>Sample</b></td>
    <td>Alpha-numeric designation (followed by _model id number)</td>
  </tr>
  <tr>
    <td valign="top"><b>Gene</b></td>
    <td>HGNC nomenclature</td>
  </tr>
  
   
  <tr>
    <td valign="top"><b>CKB Variant</b></td>
    <td>Variant annotated in CKB. Link in blue if publicly available, in red if CKB Boost subscription is required.</td>
  </tr>

  <tr>
    <td valign="top"><b>CKB Variant effect</b></td>
    <td>Variant effect as annotated in CKB.</td>
  </tr>  
  
  <tr>
    <td valign="top"><b>CKB Protein effect</b></td>
    <td>Protein effect as annotated in CKB.</td>
  </tr>

  <tr>
    <td valign="top"><b>CKB # Annotations predicting sensitivity</b></td>
    <td>The number of annotations in CKB predicting sensitivity for the variant. Shown as preclinical / clinical.</td>
  </tr>  
  <tr>
    <td valign="top"><b>CKB # Annotations predicting resistance</b></td>
    <td>The number of annotations in CKB predicting resistance for the variant. Shown as preclinical / clinical.</td>
  </tr>

  <tr>
    <td valign="top"><b>CKB Potential treatment approaches</b></td>
    <td>Potential treatment approaches identified in CKB.</td>
  </tr>  
  
  
  <tr>
    <td valign="top"><b>Platform</b></td>
    <td>Capture panel used (Truseq-JAX, CTP or Whole Exome)</td>
  </tr>
  <tr>
    <td valign="top"><b>Chromosome</b></td>
    <td>Chromosome number</td>
  </tr>
  <tr>
    <td valign="top"><b>Seq Position</b></td>
    <td>Chromosomal position of variant start (in reference sequence)</td>
  </tr>
  <tr>
    <td valign="top"><b>Ref Allele</b></td>
    <td>Nucleotide(s) present in reference sequence</td>
  </tr>
  <tr>
    <td valign="top"><b>Alt Allele</b></td>
    <td>Nucleotide(s) present in sample</td>
  </tr>
 
  <tr>
    <td valign="top"><b>Amino Acid Change</b></td>
    <td>Protein sequence change from reference</td>
  </tr>
  <tr>
    <td valign="top"><b>RS variants</b></td>
    <td>Accession numbers for public databases (dbSNP, COSMIC)</td>
  </tr>
  <tr>
    <td valign="top"><b>Read Depth</b></td>
    <td>Number of reads at variant site</td>
  </tr>
  <tr>
    <td valign="top"><b>Allele Frequency</b></td>
    <td>Percentage of variant found as part of total alleles</td>
  </tr>
  <tr>
    <td valign="top"><b>Transcript ID</b></td>
    <td>RefSeq accession for canonical transcript</td>
  </tr>
  <tr>
    <td valign="top"><b>Filtered Rationale</b></td>
    <td>Indication of filters which the variant failed.<br>
	Germline-Alt_AF_{percent} or PutativeGermline: A variant is predicted to be germline based on public databases and its alternate allele percentage frequency</td>
  </tr>
  <tr>
    <td valign="top"><b>Passage Num</b></td>
    <td>Passage number of PDX tumor sample assayed</td>
  </tr>
  <tr>
    <td valign="top"><b>Gene ID</b></td>
    <td>Accession number of gene in Entrez or Ensembl</td>
  </tr>
  
  <tr>
    <td valign="top"><b>Count Human Reads</b></td>
    <td>Total number of human reads.</td>
  </tr>
 
   <tr>
    <td valign="top"><b>PCT Human Reads</b></td>
    <td>The percentage of human reads.</td>
  </tr>
   
</table>
<br>

<a name="pdxExpression"></a><b>Expression data</b><br>

The expression data are analyzed from microarray or RNAseq platforms.<br><br>

<ul>
    <li><b>Affymetrix microarray HU133 or HG1.0ST (deprecated)</b><br>
        The arrays were processed with the AffyPLM R package, using quantile normalization,<br>
        no background correction, and fitting to a simple model that treats the log Intensity<br>
        as a sum of array effect, probe effect, and residual.<br>
        The array effect is the "summarized expression" that is equivalent to the median polished value produced by standard RMA analysis.</li>

    <li><b>RNAseq</b><br>
        RNAseq data is first processed with Xenome to extract human sequences.<br>
        The human sequences are aligned to the transcriptome with Bowtie, and then expression levels are estimated by RSEM.<br>
        RSEM estimated counts are finally upper quantile normalized.
        <br>
        <br>
        Fusion gene is analyzed using Xenome to extract human sequences and SOAPfuse.
    </li>
</ul>


Gene expression of the CTP panel genes is displayed as a chart of percentile rank z-score,<br>
which measures each gene's model-specific expression in comparison with that gene in all models assayed by the same platform.<br>
The mean and standard deviation for z-score calculation is obtained based on a fixed set of PDX samples for each platform.<br>
Other forms of expression values (e.g. z-score, normalized expression) and expression of other genes not listed on the MTB site can be made available upon request.<br>

<br>
<a name="pdxalternateloci"></a>
<b>Genes flagged with hatched bars in gene expression chart</b><br>
Some genes display a fair amount of heterogeneity in the normal population. This means that some may align poorly to the reference genome. 
The Genome Reference Consortium "provides multiple representations (alternate loci) for regions that are too complex to be represented by a single path." 
We have analyzed the data for several samples using both the primary build only and the primary build with the alternate loci. 
Using the extended reference genome introduces complications in interpreting the gene expression, so we have opted to use only the primary build. 
We are flagging the genes where the alternate loci are sufficiently different from primary to caution users that expression of these genes could be artifactually lowered.
<br>

<br>
For gene fusion, the gene symbols upstream and downstream of the fusion are reported and whether the downstream fusion partner is frame-shift or in-frame-shift.<br> 
Only those with associated drug efficacy or cancer-related evidences are reported to minimize false-positives.<br>
Other detected fusions and additional information (e.g. breakpoint coordinates) can be made available upon request.<br>
<br>
<br>
<a name="pdxCNV"></a><b>Copy number</b>
<br>
The copy number variation is analyzed from the Affymetrix Human SNP 6.0 array.<br>
PennCNV-Affy and ASCAT 2.2 is used to predict allele-specific copy number and ploidy. <br>
Gene-level copy number is obtained by intersecting copy number segments with genome coordinates of Ensembl genes.  <br>
In cases where a segment boundary is contained within a gene's coordinates, the most conservative estimate of copy number is used. <br>
<br>
In Gene CNV, the copy number of the CTP panel genes is displayed as a chart of log2(cn raw / sample ploidy).<br>
The CNV Plots present the difference from sample ploidy along the chromosomes (orange) and indicate where loss of heterozygosity occurs (blue).
<br><br>
Values for specific genes not listed on the MTB site can be made available upon request.


<br>
<br>
<a name="pdxTMB"></a><b>Tumor mutation burden (TMB) estimation:</b><br>

TMB was calculated using variants that
<ul>
    <li>(i) met all quality criteria (coverage, mapping quality etc.), </li>
    <li>(ii) were not present in an in-house curated blacklist of false positive variants from loci that prone to sequencing and analysis errors and/or 
        are associated with highly polymorphic genes (i.e., MUC4, MUC5B, MUC16, MUC17, and HLA-A </li>
    <li>(iii) are likely somatic mutations, and  </li>
    <li>(iv) have a high or moderate functional impact (i.e., non-synonymous changes, frame-shifts, stop losses/gains, and splice-site acceptor/donor changes).</li>
</ul>
   TMB was estimated by dividing the number of variants that met the criteria list above by the length (in Mb)
   of The Jackson Laboratory <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481190">Cancer Treatment Profile (CTP)</a> targeted gene panel.<br>
   <br>
   We defined high TMB as 22 mutations/Mb, which was calculated based on the TMB distribution of all PDX models analyzed as follows: Q3 (third quartile of TMB) + 1.5 x inter-quartile range of TMB.

  
<p>
<a href="#top">Back to Top</a>


<hr>
  <!--
  <a name="pdxcomparison"></a><h2>PDX Comparison Results</h2>
  <table>
    <tr><td>Rank Z based expression scale</td>
    <tr>
        <td>
            <table>
                <tr>
                <td style="text-align:center;color:#FFFFFF;background-color:#006900">-15</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#007300">-14</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#007d00">-13</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#008700">-12</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#009100">-11</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#009b00">-10</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00a500">-9</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00af00">-8</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00b900">-7</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00c300">-6</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00cd00">-5</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00d700">-4</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00e100">-3</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00eb00">-2</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#00f500">-1</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#808080"> 0 </td>
                <td style="text-align:center;color:#FFFFFF;background-color:#f50000">+1</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#eb0000">+2</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#e10000">+3</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#d70000">+4</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#cd0000">+5</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#c30000">+6</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#b90000">+7</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#af0000">+8</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#a50000">+9</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#9b0000">+10</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#910000">+11</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#870000">+12</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#7d0000">+13</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#730000">+14</td>
                <td style="text-align:center;color:#FFFFFF;background-color:#690000">+15</td>
                <td style="text-align:center;background-color:#FFFFFF">No Value</td>
                </tr>
            </table>
            
        </td>
    </tr>
    <tr><td>
            <table>
                <tr><td>CNV color coding</td></tr>
                <tr>
                    <td style="text-align:center;color:#000000;background-color:#FFA500">Amplification</td>
                    <td style="text-align:center;color:#000000;background-color:#0000FF">Deletion</td>
                    <td style="text-align:center;color:#000000;background-color:#808080">Normal</td>
                    <td style="text-align:center;color:#000000;background-color:#FFFFFF">No Value</td>
                </tr>
            </table>
        </td>
    </tr>
  </table>
            
  Click a sample name along the top of the table to go to the PDX details page for that sample.<br>
  Click on a row (gene) in the mutation section to display the descriptions of the mutations.<br>
  Click on the "+" at the top of the mutation section to display all the mutation descriptions.<br>
  Click on the "-" to hide all the mutation descriptions.
  

  

<p>
<a href="#top">Back to Top</a>
-->
<hr>
<hr>

<font size=-1><a href="citation.jsp">Citing MTB</a></font>
<br><font size=-1><a href="#top">MTB</a> is supported by grant <a href="http://commons.cit.nih.gov/crisp3/CRISP_LIB.getdoc?textkey=6434346&p_grant_num=1R01CA089713-01A1&p_query=&ticket=524183&p_audit_session_id=3455344&p_keywords=">CA89713</a>
from the National Cancer Institute (<A HREF="http://www.nci.nih.gov/">NCI</A>).</font>
<br><font size=-1><a href="copyright.jsp">Warranty Disclaimer &amp; Copyright Notice</a>.</font>
<br><font size=-1>Send questions and comments to <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a>.</font>
