<a name="top"></a>

<h2>MMHCdb Background</h2>
<ul>
	<li><a href="#data">What information can I find in the Mouse Models of Human Cancer Database?</a></li>
	<li><a href="#submitting">How can I contribute my data to MMHCdb?</a></li>
	<li><a href="#citing">How do I cite MMHCdb?</a></li>
	<li><a href="#error">What should I do if I get an error message?</a></li>
</ul>

<h2>How to query the Mouse Models of Human Cancer Database&hellip;</h2>
<ul>
	
	<li><a href="#dynamicgrid">How do I use the Tumor Frequency Grid</a></li>
</ul>



<h2>How do I interpret my search results?</h2>
<ul>
	
	<li><a href="#tumorresults">Advanced Search Results</a></li>
	<li><a href="#modeldetails">Model Details</a></li>
        <li><a href="#straindetails">Strain Details</a></li>
	<li><a href="#referencedetails">Reference Details</a></li>
	<li><a href="#pdxsearch">PDX Search</a></li>
	<li><a href="#pdxdetails">PDX Details</a></li>
        
      
	
</ul>

<h2>See Also:</h2>
<ul>
	<li><a href="http://www.informatics.jax.org/javawi2/servlet/WIFetch?page=glossaryIndex&print=no" target="_blank">MGI Glossary</a></li>
</ul>

<a href="/">MMHCdb Home Page</a>

<hr>

<a name="data"></a>
<h2>MMHCdb Background</h2>
<h3>What information can I find in the Mouse Models of Human Cancer (MMHCdb) Database?</h3>

<p>The Mouse Models of Human Cancer Database (MMHCdb) was designed to aid researchers in such areas as choosing experimental models, reviewing patterns of mutations in specific cancers, and identifying genes that are commonly mutated across a spectrum of cancers.</p>

<p>MMHCdb supports the use of the mouse as a model system of hereditary cancer by providing electronic access to:</p>

<ol>
	<li> Descriptions of tumors (data are organized by strain and organ of tumor origin)</li>
	<li>Tumor frequency and latency data</li>
	<li>Information on genetic factors associated with tumors and tumor development</li>
	<li>Tumor pathology reports and images</li>
	<li>References, both published and unpublished</li>
	<li>Links to related online resources, including:</li>
	<li><ul>
		<li><a href="http://www.informatics.jax.org/" target="_blank">Mouse Genome Informatics (MGI)</a></li>
		<li><a href="http://www.jax.org/phenome" target="_blank">Mouse Phenome Database (MPD)</a></li>
		<li><a href="http://jaxmice.jax.org/" target="_blank">JAX&reg; Mice</a></li>
		<li><a href="http://mouse.ncifcrf.gov/" target="_blank">NCI Mouse Repository</a></li>
	</ul></li>
</ol>

<h4>Important information for users of MMHCdb</h4>
<ol>
	<li><strong>MMHCdb reports negative as well as positive data.</strong> For example, strains of mice that are reported to have a zero frequency of a particular tumor type are included
	in the database.</li>
	<li><strong>Tumor data in MMHCdb are from studies of tumors that arise in mice either spontaneously or after administration of some sort of agent.</strong> MMHCdb does not include data on
	tumors that arise in humans or other species and are subsequently transplanted into mice.</li>
	<li><strong>MMHCdb is NOT current with the published literature.</strong> Populating MMHCdb is a high priority, but the volume of data available for mouse cancer biology is very large and
	it will take some time to enter the data. Descriptions of the data sources and data entry priorities for MMHCdb are described below.</li>
	<li><strong>The mouse strain information in MMHCdb is NOT limited to the strains distributed by the Jackson Laboratory.</strong> However, if a strain in MMHCdb is distributed by The Jackson
	Laboratory, a link to the data sheet in the JAX&reg; Mice database is provided.</li>
	<li><strong>The data in the Mouse Models of Human Cancer Database come from the following sources:</strong>
		<ul>
			<li>the published literature (current and retrospective),</li>
			<li>the submission of tumor pathology slides and/or images from investigators (from published or unpublished data), and</li>
			<li>tumor frequency and pathology data from routine animal health screenings of mouse colonies (often unpublished data).</li>
		</ul>
	</li>
	<li><strong>Priority for entering data in the database is determined by the novelty of the mouse model, the quality of the data, and the organ system involved.</strong> The organ systems of high priority are those with the highest reported mortality rate among the United States population, according to the National Cancer Institute.
	</li>
</ol>

<a class="jump-nav" href="#top">Back to Top</a>
<hr>

<a name="submitting"></a>
<h3>How can I contribute my data to MMHCdb?</h3>

<p>Data in the Mouse Models of Human Cancer Database come both from the published scientific literature and from unpublished, routine animal health screenings of mouse colonies. Data
submissions from the scientific community are welcome by contacting <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a> for information on the data submission process.</p>

<a class="jump-nav" href="#top">Back to Top</a>
<hr>

<a name="citing"></a>
<h3>How do I cite MMHCdb?</h3>


<p>MMHCdb is supported by grant CA89713, entitled "Electronic Access to Mouse Tumor Data", awarded to Janan T. Eppig from the National Cancer Institute (NCI) of the National Institutes
of Health (NIH).</p>

<p>Please use the following citation when referring to the Mouse Models of Human Cancer Database.
<ul><li><b>Debra M. Krupke, Dale A. Begley, John P. Sundberg, Joel E. Richardson, Steven B. Neuhauser and Carol J. Bult, The Mouse Models of Human Cancer Database: A Comprehensive Resource for Mouse Models of Human Cancer., Cancer Res October 31 2017 77 (21) e67-e70.</b></li></ul>
<p>If you wish to cite a specific area of  MMHC we suggest a format similar to the following example:
<ul><li><b>Some tumor data for this paper were retrieved from the Mouse Models of Human Cancer Database (MMHC, formerly MTB), Mouse Genome Informatics, The Jackson Laboratory, Bar Harbor, Maine. World Wide Web (URL: http://tumor.informatics.jax.org/). (February, 2019 i.e., the date you retrieved the data cited).</b></li></ul>

<p><a href="#top">Back to Top</a>
<center><hr width=80%></center>




<hr>
	
	<a name="dynamicgrid"></a><h2>Dynamic Tumor Frequency Grid</h2>
<p>The dynamic tumor frequency grid presents the same information as the existing tumor frequency grid,
	but allows the data to be refined by the user.</p>

<p>Individual strain families and organ groups can be selected to generate a customized grid.</p>

<p>In the resulting grid, individual strains and organs can be selected to further refine the grid's contents. 
	Use the check boxes to select the desired strain and organ rows and columns, then click the 'Generate Grid' button.</p>

<a href="#top">Back to Top</a>

<hr>

<a name="tumorresults"></a><h2>Advanced Search Results</h2>
<p>We have updated our user interface. New documentation is coming soon. If you need help contact us here <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a></p>
<hr>
<a name="modeldetails"></a><h2>Model Details</h2>
<p>We have updated our user interface. New documentation is coming soon. If you need help contact us here <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a></p>
<hr>
<a name="straindetails"></a><h2>Strain Details</h2>
<p>We have updated our user interface. New documentation is coming soon. If you need help contact us here <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a></p>
<hr>
<a name="referncedetails"></a><h2>Reference Details</h2>
<p>We have updated our user interface. New documentation is coming soon. If you need help contact us here <a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a></p>

<hr>
	
	<a name="pdxsearch"></a><h2>PDX Search Form</h2>
<p>Search for PDX models by primary cancer site, diagnosis, and gene variant.</p>

<strong>Tumor type</strong>

<p>The primary site and diagnosis fields are string searches and are not hierarchical.</p>

<p>So "non-small cell lung cancer" will not return "lung squamous cell carcinoma."</p>

<strong>Gene list</strong>

<p>The gene lists in the drop down menus are not comprehensive.</p>

<p>They are a subset of genes that are of interest in cancer (derived from the CTP panel).</p>

<p>The gene variant field contains any gene for which our assays find variants.</p>

<p>This field only recognizes the official HGNC gene symbol.</p>

<p>Or select a gene to see expression levels for that gene across PDX models.</p>

<p>Or select a gene to see models where that gene is amplified or deleted.</p>

<a href="#top">Back to Top</a>

<hr>
	
	<a name="pdxresults"></a><h2>PDX Search Results</h2>
<p>PDX models matching the search criteria are displayed in a dynamic table.</p>

<p>The results can be sorted by any column. Columns can be resized or hidden.</p>

<p>Click the model ID to go to the model details page to see any additional data.</p>

<p>To send an email requesting additional information on PDX models select models using the check boxes and click 'Request Details' button.</p>

<a href="#top">Back to Top</a>

<hr>
	
	<a name="pdxdetails"></a><h2>PDX Details</h2>
<p>Variant, expression and copy number data may not be available for all models.</p>

<strong>Variant data</strong>

<p>The variant data (point mutations and indels) are analyzed from next-generation sequencing using various capture panels:</p>
<ul>
		<li>Truseq (deprecated) - The Illumina TruSeq Amplicon Cancer Panel covers 48 cancer-related genes. <a href="http://www.illumina.com/products/truseq_amplicon_cancer_panel.html"> Link.</a></li>

		<li>CTP - The JAX Cancer Treatment Profile panel covers 358 cancer-associated genes <a href="https://www.jax.org/clinical-genomics/jax-cancer-treatment-profile"> Link.</a></li>

		<li>Whole Exome (limited number of samples assayed) - Agilent SureSelect human exon capture. </li>
</ul>
<p>The analysis of the sequencing output uses the Xenome tool to remove contaminating mouse sequences before alignment and variant calling.</p>

<p>BWA, GATK, and SnpEff are utilized for alignment (GRCh38 human reference), variant calling and annotation.</p>

<p>The variant results also include clinical annotations using the JAX proprietary clinical knowledge base (CKB).</p>

<p>The high quality variants are further filtered to remove likely false positives,</p>

<p>variants with low and modifier impact based on SnpEff annotations, and putative germline variants.</p>

<p>However, all high quality variants that are listed in the JAX CKB database with associated drug efficacy</p>

<p>or cancer-related evidences are retained regardless of the filters. 

	The following table describes the fields used in the variant summary data.</p>

<table>
	<tr>
		<th>Field</th>
		<th>Description</th>
	</tr>
	<tr>
		<td><h4>Model</h4></td>
		<td><p>'T' or 'J' number ID of model</p></td>
	</tr>
	<tr>
		<td><h4>Sample</h4></td>
		<td><p>Alpha-numeric designation (followed by _model id number)</p></td>
	</tr>
	<tr>
		<td><h4>Gene</h4></td>
		<td><p>HGNC nomenclature</p></td>
	</tr>
	<tr>
		<td><h4>Platform</h4></td>
		<td><p>Capture panel used (Truseq-JAX, CTP or Whole Exome)</p></td>
	</tr>
	<tr>
		<td><h4>Chromosome</h4></td>
		<td><p>Chromosome number</p></td>
	</tr>
	<tr>
		<td><h4>Seq Position</h4></td>
		<td><p>Chromosomal position of variant start (in reference sequence)</p></td>
	</tr>
	<tr>
		<td><h4>Ref Allele</h4></td>
		<td><p>Nucleotide(s) present in reference sequence</p></td>
	</tr>
	<tr>
		<td><h4>Alt Allele</h4></td>
		<td><p>Nucleotide(s) present in sample</p></td>
	</tr>
	<tr>
		<td><h4>Consequence</h4></td>
		<td><p>Functional annotation of the variant</p></td>
	</tr>
	<tr>
		<td><h4>Amino Acid Change</h4></td>
		<td><p>Protein sequence change from reference</p></td>
	</tr>
	<tr>
		<td><h4>RS variants</h4></td>
		<td><p>Accession numbers for public databases (dbSNP, COSMIC)</p></td>
	</tr>
	<tr>
		<td><h4>Read Depth</h4></td>
		<td><p>Number of reads at variant site</p></td>
	</tr>
	<tr>
		<td><h4>Allele Frequency</h4></td>
		<td><p>Percentage of variant found as part of total alleles</p></td>
	</tr>
	<tr>
		<td><h4>Transcript ID</h4></td>
		<td><p>RefSeq accession for canonical transcript</p></td>
	</tr>
	<tr>
		<td><h4>Filtered Rationale</h4></td>
		<td><p>Indication of filters which the variant failed.</p>

<p>Germline-Alt_AF_{percent} or PutativeGermline: A variant is predicted to be germline based on public databases and its alternate allele percentage frequency</p>
</td>
	</tr>
	<tr>
		<td><h4>Passage Num</h4></td>
		<td><p>Passage number of PDX tumor sample assayed</p></td>
	</tr>
	<tr>
		<td><h4>Gene ID</h4></td>
		<td><p>Accession number of gene in Entrez or Ensembl</p></td>
	</tr>
	<tr>
		<td><h4>CKB Evidence</h4></td>
		<td><p>Types of evidence in JAX CKB associated with the variant.</p>

<p>Actionable: Actionable evidence is clinical or preclinical data supporting a connection between a gene variant and a drug response.</p>

<p>The related response type may be sensitive or resistant.</p>

<p>Emerging: Emerging provides evidence for potential development of a gene variant as a future cancer therapy target.</p>
</td>
	</tr>
	<tr>
		<td><h4>Actionable Cancer Types</h4></td>
		<td><p>Tumor type directly associated with the variant in JAX CKB</p></td>
	</tr>
	 <tr>
		<td><h4>Drug Class</h4></td>
		<td><p>Drug classes of treatment approach associated with the variant in JAX CKB</p></td>
	</tr>
	 <tr>
		<td><h4>Variant Num Trials</h4></td>
		<td><p>Number of clinical trials directly associated with the variant (The clinical trials associated with the gene can be made available upon request.)</p></td>
	</tr>
	 <tr>
		<td><h4>Variant NCT IDs</h4></td>
		<td><p>NCT IDs of clinical trials directly associated with the variant </p></td>
	</tr>
</table>

<strong>Expression data</strong>

<p>The expression data are analyzed from microarray or RNAseq platforms.</p>

<ul>
		<li><h4>Affymetrix microarray HU133 or HG1.0ST (deprecated)</h4>

<p>The arrays were processed with the AffyPLM R package, using quantile normalization,</p>

<p>no background correction, and fitting to a simple model that treats the log Intensity</p>

<p>as a sum of array effect, probe effect, and residual.</p>

<p>The array effect is the "summarized expression" that is equivalent to the median polished value produced by standard RMA analysis.</p>
</li>

		<li><h4>RNAseq</h4>

<p>RNAseq data is first processed with Xenome to extract human sequences.</p>

<p>The human sequences are aligned to the transcriptome with Bowtie, and then expression levels are estimated by RSEM.</p>

<p>RSEM estimated counts are finally upper quantile normalized.</p>

<p>Fusion gene is analyzed using Xenome to extract human sequences and SOAPfuse.</p>
</li>
</ul>
<p>Gene expression of the CTP panel genes is displayed as a chart of percentile rank z-score,</p>

<p>which measures each gene's model-specific expression in comparison with that gene in all models assayed by the same platform.</p>

<p>The mean and standard deviation for z-score calculation is obtained based on a fixed set of PDX samples for each platform.</p>

<p>Other forms of expression values (e.g. z-score, normalized expression) and expression of other genes not listed on the MMHCdb site can be made available upon request.</p>

<a name="pdxalternateloci"></a>
<h4>Genes flagged with hatched bars in gene expression chart</h4>

<p>Some genes display a fair amount of heterogeneity in the normal population. This means that some may align poorly to the reference genome. 
The Genome Reference Consortium "provides multiple representations (alternate loci) for regions that are too complex to be represented by a single path." 
We have analyzed the data for several samples using both the primary build only and the primary build with the alternate loci. 
Using the extended reference genome introduces complications in interpreting the gene expression, so we have opted to use only the primary build.</p>

<p>We are flagging the genes where the alternate loci are sufficiently different from primary to caution users that expression of these genes could be artifactually lowered.</p>

<p>For gene fusion, the gene symbols upstream and downstream of the fusion are reported and whether the downstream fusion partner is frame-shift or in-frame-shift. Only those with associated drug efficacy or cancer-related evidences are reported to minimize false-positives. Other detected fusions and additional information (e.g. breakpoint coordinates) can be made available upon request.</p>

<a name="pdxCNV"></a>
<h4>Copy number</h4>

<p>The copy number variation is analyzed from the Affymetrix Human SNP 6.0 array. PennCNV-Affy and ASCAT 2.2 is used to predict allele-specific copy number and ploidy. Gene-level copy number is obtained by intersecting copy number segments with genome coordinates of Ensembl genes. In cases where a segment boundary is contained within a gene's coordinates, the most conservative estimate of copy number is used.</p>


<p>In Gene CNV, the copy number of the CTP panel genes is displayed as a chart of log2(cn raw / sample ploidy). The CNV Plots present the difference from sample ploidy along the chromosomes (orange) and indicate where loss of heterozygosity occurs (blue).</p>


<p>Values for specific genes not listed on the MMHCdb site can be made available upon request.</p>


<a name="pdxTMB"></a>
<h4>Tumor mutation burden (TMB) estimation:</h4>


<p>TMB was calculated using variants that</p>
<ul>
    <li>(i) met all quality criteria (coverage, mapping quality etc.), </li>
    <li>(ii) were not present in an in-house curated blacklist of false positive variants from loci that prone to sequencing and analysis errors and/or 
        are associated with highly polymorphic genes (i.e., MUC4, MUC5B, MUC16, MUC17, and HLA-A </li>
    <li>(iii) are likely somatic mutations, and  </li>
    <li>(iv) have a high or moderate functional impact (i.e., non-synonymous changes, frame-shifts, stop losses/gains, and splice-site acceptor/donor changes).</li>
</ul>


  <p> TMB was estimated by dividing the number of variants that met the criteria list above by the length (in Mb) of The Jackson Laboratory <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4481190">Cancer Treatment Profile (CTP)</a> targeted gene panel.</p>
   

   <p>We defined high TMB as 22 mutations/Mb, which was calculated based on the TMB distribution of all PDX models analyzed as follows: Q3 (third quartile of TMB) + 1.5 x inter-quartile range of TMB.</p>


<a name="pdxMSI"></a>
<h4>Microsatellite Instability (MSI)</h4>

<p>The <a href="http://tumor.informatics.jax.org/mtbwi/live/www/html/MSIDocumentation.pdf" target="_blank">MSIsensor2</a> algorithm was used to determine MSI status of JAX samples. The samples with MSI Score (MSI-Percentage > 20%)  are considered at MSI-H. This threshold demonstrates good differentiation between MSI-High (MSI-H) and MSI-Stable (MSI-S) samples during MSIsensor2 algorithm development and our internal benchmarking. </p>


<a href="#top">Back to Top</a>

<p>Detected fusion genes are annotated with the clinical information from the JAX CKB database (similar to variant annotation described above),
and only those with associated drug efficacy or cancer-related evidences are reported to minimize false-positives.</p>

<p>Other detected fusions and additional information (e.g. breakpoint coordinates) can be made available upon request.</p>

<h4>Copy number</h4>

<p>The copy number variation is analyzed from the Affymetrix Human SNP 6.0 array.</p>

<p>PennCNV-Affy and ASCAT 2.2 is used to predict allele-specific copy number and ploidy.</p>

<p>Gene-level copy number is obtained by intersecting copy number segments with genome coordinates of Ensembl genes.</p>

<p>In cases where a segment boundary is contained within a gene's coordinates, the most conservative estimate of copy number is used.</p>

<p>The copy number of the CTP panel genes is displayed as a chart of log2(cn raw / sample ploidy).</p>

<p>Other forms of copy number values (e.g. gain and loss relative to ploidy, loss of heterozygosity)</p>

<p>and copy number of other genes not listed on the MMHCdb site can be made available upon request.</p>

<a href="#top">Back to Top</a>

<hr>
	
	<a name="pdxcomparison"></a><h2>PDX Comparison Results</h2>
	<table>
		<tr><td><p>Rank Z based expression scale</p></td>
		<tr>
				<td>
						<table class="pdx-result-expression">
								<tr>
								<td style="background-color:#006900">-15</td>
								<td style="background-color:#007300">-14</td>
								<td style="background-color:#007d00">-13</td>
								<td style="background-color:#008700">-12</td>
								<td style="background-color:#009100">-11</td>
								<td style="background-color:#009b00">-10</td>
								<td style="background-color:#00a500">-9</td>
								<td style="background-color:#00af00">-8</td>
								<td style="background-color:#00b900">-7</td>
								<td style="background-color:#00c300">-6</td>
								<td style="background-color:#00cd00">-5</td>
								<td style="background-color:#00d700">-4</td>
								<td style="background-color:#00e100">-3</td>
								<td style="background-color:#00eb00">-2</td>
								<td style="background-color:#00f500">-1</td>
								<td style="background-color:#808080"> 0 </td>
								<td style="background-color:#f50000">+1</td>
								<td style="background-color:#eb0000">+2</td>
								<td style="background-color:#e10000">+3</td>
								<td style="background-color:#d70000">+4</td>
								<td style="background-color:#cd0000">+5</td>
								<td style="background-color:#c30000">+6</td>
								<td style="background-color:#b90000">+7</td>
								<td style="background-color:#af0000">+8</td>
								<td style="background-color:#a50000">+9</td>
								<td style="background-color:#9b0000">+10</td>
								<td style="background-color:#910000">+11</td>
								<td style="background-color:#870000">+12</td>
								<td style="background-color:#7d0000">+13</td>
								<td style="background-color:#730000">+14</td>
								<td style="background-color:#690000">+15</td>
								<td style="text-align:center;background-color:#FFFFFF">No Value</td>
								</tr>
						</table>
						
				</td>
		</tr>
		<tr><td>
						<table class="pdx-result-cnv">
								<tr><td><p>CNV color coding</p></td></tr>
								<tr>
										<td style="background-color:#FFA500">Amplification</td>
										<td style="background-color:#0000FF">Deletion</td>
										<td style="background-color:#808080">Normal</td>
										<td style="background-color:#FFFFFF">No Value</td>
								</tr>
						</table>
				</td>
		</tr>
	</table>
<p>Click a sample name along the top of the table to go to the PDX details page for that sample.</p>

<p>Click on a row (gene) in the mutation section to display the descriptions of the mutations.</p>

<p>Click on the "+" at the top of the mutation section to display all the mutation descriptions.</p>

<p>Click on the "-" to hide all the mutation descriptions.</p>

<a href="#top">Back to Top</a>

<hr>

<p><a href="citation.jsp">Citing MMHCdb</a></p>

<p><a href="#top">MMHCdb</a> is supported by grant <a href="http://commons.cit.nih.gov/crisp3/CRISP_LIB.getdoc?textkey=6434346&p_grant_num=1R01CA089713-01A1&p_query=&ticket=524183&p_audit_session_id=3455344&p_keywords=">CA89713</a> from the National Cancer Institute (<A HREF="http://www.nci.nih.gov/">NCI</a>).</p>

<p><a href="copyright.jsp">Warranty Disclaimer &amp; Copyright Notice</a>

<p>Send questions and comments to</p>
<a href="http://www.informatics.jax.org/mgihome/support/mgi_inbox.shtml">User Support</a>
